// @zernio/node/dist/index.mjs
var __defProp = Object.defineProperty;
var __defNormalProp = (obj, key, value) => key in obj ? __defProp(obj, key, { enumerable: true, configurable: true, writable: true, value }) : obj[key] = value;
var __publicField = (obj, key, value) => __defNormalProp(obj, typeof key !== "symbol" ? key + "" : key, value);

// package.json
var package_default = {
  name: "@zernio/node",
  version: "0.2.378",
  description: "The official Node.js library for the Zernio API",
  main: "dist/index.js",
  module: "dist/index.mjs",
  types: "dist/index.d.ts",
  sideEffects: false,
  exports: {
    ".": {
      types: "./dist/index.d.ts",
      import: "./dist/index.mjs",
      require: "./dist/index.js"
    },
    "./_shims/*": {
      types: "./dist/_shims/*.d.ts",
      import: "./dist/_shims/*.mjs",
      require: "./dist/_shims/*.js"
    }
  },
  files: [
    "dist",
    "src",
    "README.md",
    "LICENSE"
  ],
  scripts: {
    generate: "openapi-ts -i openapi.yaml -o src/generated -c @hey-api/client-fetch && npm run generate:client",
    "generate:client": "npx tsx scripts/generate-client.ts",
    build: "tsup src/index.ts --format cjs,esm --dts --clean",
    dev: "tsup src/index.ts --format cjs,esm --dts --watch",
    lint: "eslint src",
    typecheck: "tsc --noEmit",
    test: "vitest run",
    "test:watch": "vitest",
    prepublishOnly: "npm run build",
    "fetch-spec": "curl -o openapi.yaml https://zernio.com/openapi.yaml"
  },
  keywords: [
    "zernio",
    "social-media",
    "scheduling",
    "api",
    "sdk",
    "instagram",
    "tiktok",
    "youtube",
    "linkedin",
    "twitter",
    "x",
    "facebook",
    "pinterest",
    "threads",
    "bluesky",
    "snapchat",
    "telegram"
  ],
  author: "Zernio <support@zernio.com>",
  license: "Apache-2.0",
  repository: {
    type: "git",
    url: "git+https://github.com/zernio-dev/zernio-node.git"
  },
  homepage: "https://zernio.com/docs/api",
  bugs: {
    url: "https://github.com/zernio-dev/zernio-node/issues"
  },
  engines: {
    node: ">=18"
  },
  devDependencies: {
    "@hey-api/client-fetch": "^0.6.0",
    "@hey-api/openapi-ts": "^0.61.0",
    "@types/node": "^20.11.0",
    "@typescript-eslint/eslint-plugin": "^7.18.0",
    "@typescript-eslint/parser": "^7.18.0",
    eslint: "^8.56.0",
    tsup: "^8.0.1",
    tsx: "^4.7.0",
    typescript: "^5.3.3",
    vitest: "^1.2.0",
    yaml: "^2.3.4"
  }
};

// node_modules/@hey-api/client-fetch/dist/index.js
var U = /\{[^{}]+\}/g;
var S = ({ allowReserved: t, name: r, value: e }) => {
  if (e == null) return "";
  if (typeof e == "object") throw new Error("Deeply-nested arrays/objects aren\u2019t supported. Provide your own `querySerializer()` to handle these.");
  return `${r}=${t ? e : encodeURIComponent(e)}`;
};
var $ = (t) => {
  switch (t) {
    case "label":
      return ".";
    case "matrix":
      return ";";
    case "simple":
      return ",";
    default:
      return "&";
  }
};
var k = (t) => {
  switch (t) {
    case "form":
      return ",";
    case "pipeDelimited":
      return "|";
    case "spaceDelimited":
      return "%20";
    default:
      return ",";
  }
};
var D = (t) => {
  switch (t) {
    case "label":
      return ".";
    case "matrix":
      return ";";
    case "simple":
      return ",";
    default:
      return "&";
  }
};
var j = ({ allowReserved: t, explode: r, name: e, style: s, value: o }) => {
  if (!r) {
    let n = (t ? o : o.map((c) => encodeURIComponent(c))).join(k(s));
    switch (s) {
      case "label":
        return `.${n}`;
      case "matrix":
        return `;${e}=${n}`;
      case "simple":
        return n;
      default:
        return `${e}=${n}`;
    }
  }
  let a = $(s), i = o.map((n) => s === "label" || s === "simple" ? t ? n : encodeURIComponent(n) : S({ allowReserved: t, name: e, value: n })).join(a);
  return s === "label" || s === "matrix" ? a + i : i;
};
var A = ({ allowReserved: t, explode: r, name: e, style: s, value: o }) => {
  if (o instanceof Date) return `${e}=${o.toISOString()}`;
  if (s !== "deepObject" && !r) {
    let n = [];
    Object.entries(o).forEach(([f, p]) => {
      n = [...n, f, t ? p : encodeURIComponent(p)];
    });
    let c = n.join(",");
    switch (s) {
      case "form":
        return `${e}=${c}`;
      case "label":
        return `.${c}`;
      case "matrix":
        return `;${e}=${c}`;
      default:
        return c;
    }
  }
  let a = D(s), i = Object.entries(o).map(([n, c]) => S({ allowReserved: t, name: s === "deepObject" ? `${e}[${n}]` : n, value: c })).join(a);
  return s === "label" || s === "matrix" ? a + i : i;
};
var _ = ({ path: t, url: r }) => {
  let e = r, s = r.match(U);
  if (s) for (let o of s) {
    let a = false, i = o.substring(1, o.length - 1), n = "simple";
    i.endsWith("*") && (a = true, i = i.substring(0, i.length - 1)), i.startsWith(".") ? (i = i.substring(1), n = "label") : i.startsWith(";") && (i = i.substring(1), n = "matrix");
    let c = t[i];
    if (c == null) continue;
    if (Array.isArray(c)) {
      e = e.replace(o, j({ explode: a, name: i, style: n, value: c }));
      continue;
    }
    if (typeof c == "object") {
      e = e.replace(o, A({ explode: a, name: i, style: n, value: c }));
      continue;
    }
    if (n === "matrix") {
      e = e.replace(o, `;${S({ name: i, value: c })}`);
      continue;
    }
    let f = encodeURIComponent(n === "label" ? `.${c}` : c);
    e = e.replace(o, f);
  }
  return e;
};
var C = ({ allowReserved: t, array: r, object: e } = {}) => (o) => {
  let a = [];
  if (o && typeof o == "object") for (let i in o) {
    let n = o[i];
    if (n != null) {
      if (Array.isArray(n)) {
        a = [...a, j({ allowReserved: t, explode: true, name: i, style: "form", value: n, ...r })];
        continue;
      }
      if (typeof n == "object") {
        a = [...a, A({ allowReserved: t, explode: true, name: i, style: "deepObject", value: n, ...e })];
        continue;
      }
      a = [...a, S({ allowReserved: t, name: i, value: n })];
    }
  }
  return a.join("&");
};
var w = (t) => {
  if (!t) return "stream";
  let r = t.split(";")[0]?.trim();
  if (r) {
    if (r.startsWith("application/json") || r.endsWith("+json")) return "json";
    if (r === "multipart/form-data") return "formData";
    if (["application/", "audio/", "image/", "video/"].some((e) => r.startsWith(e))) return "blob";
    if (r.startsWith("text/")) return "text";
  }
};
var H = async (t, r) => {
  let e = typeof r == "function" ? await r(t) : r;
  if (e) return t.scheme === "bearer" ? `Bearer ${e}` : t.scheme === "basic" ? `Basic ${btoa(e)}` : e;
};
var P = async ({ security: t, ...r }) => {
  for (let e of t) {
    let s = await H(e, r.auth);
    if (!s) continue;
    let o = e.name ?? "Authorization";
    switch (e.in) {
      case "query":
        r.query || (r.query = {}), r.query[o] = s;
        break;
      case "header":
      default:
        r.headers.set(o, s);
        break;
    }
    return;
  }
};
var b = (t) => B({ baseUrl: t.baseUrl ?? "", path: t.path, query: t.query, querySerializer: typeof t.querySerializer == "function" ? t.querySerializer : C(t.querySerializer), url: t.url });
var B = ({ baseUrl: t, path: r, query: e, querySerializer: s, url: o }) => {
  let a = o.startsWith("/") ? o : `/${o}`, i = t + a;
  r && (i = _({ path: r, url: i }));
  let n = e ? s(e) : "";
  return n.startsWith("?") && (n = n.substring(1)), n && (i += `?${n}`), i;
};
var R = (t, r) => {
  let e = { ...t, ...r };
  return e.baseUrl?.endsWith("/") && (e.baseUrl = e.baseUrl.substring(0, e.baseUrl.length - 1)), e.headers = O(t.headers, r.headers), e;
};
var O = (...t) => {
  let r = new Headers();
  for (let e of t) {
    if (!e || typeof e != "object") continue;
    let s = e instanceof Headers ? e.entries() : Object.entries(e);
    for (let [o, a] of s) if (a === null) r.delete(o);
    else if (Array.isArray(a)) for (let i of a) r.append(o, i);
    else a !== void 0 && r.set(o, typeof a == "object" ? JSON.stringify(a) : a);
  }
  return r;
};
var y = class {
  constructor() {
    __publicField(this, "_fns");
    this._fns = [];
  }
  clear() {
    this._fns = [];
  }
  exists(r) {
    return this._fns.indexOf(r) !== -1;
  }
  eject(r) {
    let e = this._fns.indexOf(r);
    e !== -1 && (this._fns = [...this._fns.slice(0, e), ...this._fns.slice(e + 1)]);
  }
  use(r) {
    this._fns = [...this._fns, r];
  }
};
var E = () => ({ error: new y(), request: new y(), response: new y() });
var x = (t, r, e) => {
  typeof e == "string" || e instanceof Blob ? t.append(r, e) : t.append(r, JSON.stringify(e));
};
var W = { bodySerializer: (t) => {
  let r = new FormData();
  return Object.entries(t).forEach(([e, s]) => {
    s != null && (Array.isArray(s) ? s.forEach((o) => x(r, e, o)) : x(r, e, s));
  }), r;
} };
var I = { bodySerializer: (t) => JSON.stringify(t) };
var Q = C({ allowReserved: false, array: { explode: true, style: "form" }, object: { explode: true, style: "deepObject" } });
var V = { "Content-Type": "application/json" };
var q = (t = {}) => ({ ...I, baseUrl: "", headers: V, parseAs: "auto", querySerializer: Q, ...t });
var M = (t = {}) => {
  let r = R(q(), t), e = () => ({ ...r }), s = (i) => (r = R(r, i), e()), o = E(), a = async (i) => {
    let n = { ...r, ...i, fetch: i.fetch ?? r.fetch ?? globalThis.fetch, headers: O(r.headers, i.headers) };
    n.security && await P({ ...n, security: n.security }), n.body && n.bodySerializer && (n.body = n.bodySerializer(n.body)), n.body || n.headers.delete("Content-Type");
    let c = b(n), f = { redirect: "follow", ...n }, p = new Request(c, f);
    for (let u of o.request._fns) p = await u(p, n);
    let T = n.fetch, l = await T(p);
    for (let u of o.response._fns) l = await u(l, p, n);
    let h = { request: p, response: l };
    if (l.ok) {
      if (l.status === 204 || l.headers.get("Content-Length") === "0") return { data: {}, ...h };
      let u = (n.parseAs === "auto" ? w(l.headers.get("Content-Type")) : n.parseAs) ?? "json";
      if (u === "stream") return { data: l.body, ...h };
      let g = await l[u]();
      return u === "json" && (n.responseValidator && await n.responseValidator(g), n.responseTransformer && (g = await n.responseTransformer(g))), { data: g, ...h };
    }
    let m = await l.text();
    try {
      m = JSON.parse(m);
    } catch {
    }
    let d = m;
    for (let u of o.error._fns) d = await u(m, l, p, n);
    if (d = d || {}, n.throwOnError) throw d;
    return { error: d, ...h };
  };
  return { buildUrl: b, connect: (i) => a({ ...i, method: "CONNECT" }), delete: (i) => a({ ...i, method: "DELETE" }), get: (i) => a({ ...i, method: "GET" }), getConfig: e, head: (i) => a({ ...i, method: "HEAD" }), interceptors: o, options: (i) => a({ ...i, method: "OPTIONS" }), patch: (i) => a({ ...i, method: "PATCH" }), post: (i) => a({ ...i, method: "POST" }), put: (i) => a({ ...i, method: "PUT" }), request: a, setConfig: s, trace: (i) => a({ ...i, method: "TRACE" }) };
};

// src/generated/sdk.gen.ts
var client = M(q());
var validatePostLength = (options) => {
  return (options?.client ?? client).post({
    ...options,
    url: "/v1/tools/validate/post-length"
  });
};
var validatePost = (options) => {
  return (options?.client ?? client).post({
    ...options,
    url: "/v1/tools/validate/post"
  });
};
var validateMedia = (options) => {
  return (options?.client ?? client).post({
    ...options,
    url: "/v1/tools/validate/media"
  });
};
var validateSubreddit = (options) => {
  return (options?.client ?? client).get({
    ...options,
    url: "/v1/tools/validate/subreddit"
  });
};
var getAnalytics = (options) => {
  return (options?.client ?? client).get({
    ...options,
    url: "/v1/analytics"
  });
};
var getYouTubeChannelInsights = (options) => {
  return (options?.client ?? client).get({
    ...options,
    url: "/v1/analytics/youtube/channel-insights"
  });
};
var getLinkedInOrgAggregateAnalytics = (options) => {
  return (options?.client ?? client).get({
    ...options,
    url: "/v1/analytics/linkedin/org-aggregate-analytics"
  });
};
var getTikTokAccountInsights = (options) => {
  return (options?.client ?? client).get({
    ...options,
    url: "/v1/analytics/tiktok/account-insights"
  });
};
var getYouTubeDailyViews = (options) => {
  return (options?.client ?? client).get({
    ...options,
    url: "/v1/analytics/youtube/daily-views"
  });
};
var getYouTubeVideoRetention = (options) => {
  return (options?.client ?? client).get({
    ...options,
    url: "/v1/analytics/youtube/video-retention"
  });
};
var getFacebookPageInsights = (options) => {
  return (options?.client ?? client).get({
    ...options,
    url: "/v1/analytics/facebook/page-insights"
  });
};
var getInstagramAccountInsights = (options) => {
  return (options?.client ?? client).get({
    ...options,
    url: "/v1/analytics/instagram/account-insights"
  });
};
var getInstagramFollowerHistory = (options) => {
  return (options?.client ?? client).get({
    ...options,
    url: "/v1/analytics/instagram/follower-history"
  });
};
var getInstagramDemographics = (options) => {
  return (options?.client ?? client).get({
    ...options,
    url: "/v1/analytics/instagram/demographics"
  });
};
var getYouTubeDemographics = (options) => {
  return (options?.client ?? client).get({
    ...options,
    url: "/v1/analytics/youtube/demographics"
  });
};
var getDailyMetrics = (options) => {
  return (options?.client ?? client).get({
    ...options,
    url: "/v1/analytics/daily-metrics"
  });
};
var getBestTimeToPost = (options) => {
  return (options?.client ?? client).get({
    ...options,
    url: "/v1/analytics/best-time"
  });
};
var getContentDecay = (options) => {
  return (options?.client ?? client).get({
    ...options,
    url: "/v1/analytics/content-decay"
  });
};
var getPostingFrequency = (options) => {
  return (options?.client ?? client).get({
    ...options,
    url: "/v1/analytics/posting-frequency"
  });
};
var getPostTimeline = (options) => {
  return (options?.client ?? client).get({
    ...options,
    url: "/v1/analytics/post-timeline"
  });
};
var getGoogleBusinessPerformance = (options) => {
  return (options?.client ?? client).get({
    ...options,
    url: "/v1/analytics/googlebusiness/performance"
  });
};
var getGoogleBusinessSearchKeywords = (options) => {
  return (options?.client ?? client).get({
    ...options,
    url: "/v1/analytics/googlebusiness/search-keywords"
  });
};
var getInboxVolume = (options) => {
  return (options?.client ?? client).get({
    ...options,
    url: "/v1/analytics/inbox/volume"
  });
};
var getInboxHeatmap = (options) => {
  return (options?.client ?? client).get({
    ...options,
    url: "/v1/analytics/inbox/heatmap"
  });
};
var getInboxSourceBreakdown = (options) => {
  return (options?.client ?? client).get({
    ...options,
    url: "/v1/analytics/inbox/source-breakdown"
  });
};
var getInboxResponseTime = (options) => {
  return (options?.client ?? client).get({
    ...options,
    url: "/v1/analytics/inbox/response-time"
  });
};
var getInboxTopAccounts = (options) => {
  return (options?.client ?? client).get({
    ...options,
    url: "/v1/analytics/inbox/top-accounts"
  });
};
var listInboxConversationAnalytics = (options) => {
  return (options?.client ?? client).get({
    ...options,
    url: "/v1/analytics/inbox/conversations"
  });
};
var getInboxConversationAnalytics = (options) => {
  return (options?.client ?? client).get({
    ...options,
    url: "/v1/analytics/inbox/conversations/{conversationId}"
  });
};
var listAccountGroups = (options) => {
  return (options?.client ?? client).get({
    ...options,
    url: "/v1/account-groups"
  });
};
var createAccountGroup = (options) => {
  return (options?.client ?? client).post({
    ...options,
    url: "/v1/account-groups"
  });
};
var updateAccountGroup = (options) => {
  return (options?.client ?? client).put({
    ...options,
    url: "/v1/account-groups/{groupId}"
  });
};
var deleteAccountGroup = (options) => {
  return (options?.client ?? client).delete({
    ...options,
    url: "/v1/account-groups/{groupId}"
  });
};
var getMediaPresignedUrl = (options) => {
  return (options?.client ?? client).post({
    ...options,
    url: "/v1/media/presign"
  });
};
var searchReddit = (options) => {
  return (options?.client ?? client).get({
    ...options,
    url: "/v1/reddit/search"
  });
};
var getRedditFeed = (options) => {
  return (options?.client ?? client).get({
    ...options,
    url: "/v1/reddit/feed"
  });
};
var getBilling = (options) => {
  return (options?.client ?? client).get({
    ...options,
    url: "/v1/billing"
  });
};
var getXApiPricing = (options) => {
  return (options?.client ?? client).get({
    ...options,
    url: "/v1/billing/x-pricing"
  });
};
var getUsage = (options) => {
  return (options?.client ?? client).get({
    ...options,
    url: "/v1/usage"
  });
};
var getUsageStats = (options) => {
  return (options?.client ?? client).get({
    ...options,
    url: "/v1/usage-stats"
  });
};
var getCallsUsage = (options) => {
  return (options?.client ?? client).get({
    ...options,
    url: "/v1/usage/calls"
  });
};
var getSmsUsage = (options) => {
  return (options?.client ?? client).get({
    ...options,
    url: "/v1/usage/sms"
  });
};
var listPosts = (options) => {
  return (options?.client ?? client).get({
    ...options,
    url: "/v1/posts"
  });
};
var createPost = (options) => {
  return (options?.client ?? client).post({
    ...options,
    url: "/v1/posts"
  });
};
var syncExternalPosts = (options) => {
  return (options?.client ?? client).post({
    ...options,
    url: "/v1/posts/sync-external"
  });
};
var getPost = (options) => {
  return (options?.client ?? client).get({
    ...options,
    url: "/v1/posts/{postId}"
  });
};
var updatePost = (options) => {
  return (options?.client ?? client).put({
    ...options,
    url: "/v1/posts/{postId}"
  });
};
var deletePost = (options) => {
  return (options?.client ?? client).delete({
    ...options,
    url: "/v1/posts/{postId}"
  });
};
var bulkUploadPosts = (options) => {
  return (options?.client ?? client).post({
    ...options,
    ...W,
    headers: {
      "Content-Type": null,
      ...options?.headers
    },
    url: "/v1/posts/bulk-upload"
  });
};
var retryPost = (options) => {
  return (options?.client ?? client).post({
    ...options,
    url: "/v1/posts/{postId}/retry"
  });
};
var unpublishPost = (options) => {
  return (options?.client ?? client).post({
    ...options,
    url: "/v1/posts/{postId}/unpublish"
  });
};
var editPost = (options) => {
  return (options?.client ?? client).post({
    ...options,
    url: "/v1/posts/{postId}/edit"
  });
};
var updatePostMetadata = (options) => {
  return (options?.client ?? client).post({
    ...options,
    url: "/v1/posts/{postId}/update-metadata"
  });
};
var listUsers = (options) => {
  return (options?.client ?? client).get({
    ...options,
    url: "/v1/users"
  });
};
var getUser = (options) => {
  return (options?.client ?? client).get({
    ...options,
    url: "/v1/users/{userId}"
  });
};
var listProfiles = (options) => {
  return (options?.client ?? client).get({
    ...options,
    url: "/v1/profiles"
  });
};
var createProfile = (options) => {
  return (options?.client ?? client).post({
    ...options,
    url: "/v1/profiles"
  });
};
var getProfile = (options) => {
  return (options?.client ?? client).get({
    ...options,
    url: "/v1/profiles/{profileId}"
  });
};
var updateProfile = (options) => {
  return (options?.client ?? client).put({
    ...options,
    url: "/v1/profiles/{profileId}"
  });
};
var deleteProfile = (options) => {
  return (options?.client ?? client).delete({
    ...options,
    url: "/v1/profiles/{profileId}"
  });
};
var listAccounts = (options) => {
  return (options?.client ?? client).get({
    ...options,
    url: "/v1/accounts"
  });
};
var getFollowerStats = (options) => {
  return (options?.client ?? client).get({
    ...options,
    url: "/v1/accounts/follower-stats"
  });
};
var updateAccount = (options) => {
  return (options?.client ?? client).put({
    ...options,
    url: "/v1/accounts/{accountId}"
  });
};
var moveAccountToProfile = (options) => {
  return (options?.client ?? client).patch({
    ...options,
    url: "/v1/accounts/{accountId}"
  });
};
var deleteAccount = (options) => {
  return (options?.client ?? client).delete({
    ...options,
    url: "/v1/accounts/{accountId}"
  });
};
var getAllAccountsHealth = (options) => {
  return (options?.client ?? client).get({
    ...options,
    url: "/v1/accounts/health"
  });
};
var getAccountHealth = (options) => {
  return (options?.client ?? client).get({
    ...options,
    url: "/v1/accounts/{accountId}/health"
  });
};
var getTikTokCreatorInfo = (options) => {
  return (options?.client ?? client).get({
    ...options,
    url: "/v1/accounts/{accountId}/tiktok/creator-info"
  });
};
var listApiKeys = (options) => {
  return (options?.client ?? client).get({
    ...options,
    url: "/v1/api-keys"
  });
};
var createApiKey = (options) => {
  return (options?.client ?? client).post({
    ...options,
    url: "/v1/api-keys"
  });
};
var deleteApiKey = (options) => {
  return (options?.client ?? client).delete({
    ...options,
    url: "/v1/api-keys/{keyId}"
  });
};
var createInviteToken = (options) => {
  return (options?.client ?? client).post({
    ...options,
    url: "/v1/invite/tokens"
  });
};
var getConnectUrl = (options) => {
  return (options?.client ?? client).get({
    ...options,
    url: "/v1/connect/{platform}"
  });
};
var handleOAuthCallback = (options) => {
  return (options?.client ?? client).post({
    ...options,
    url: "/v1/connect/{platform}"
  });
};
var connectAds = (options) => {
  return (options?.client ?? client).get({
    ...options,
    url: "/v1/connect/{platform}/ads"
  });
};
var configureTikTokAdsBrandIdentity = (options) => {
  return (options?.client ?? client).patch({
    ...options,
    url: "/v1/connect/tiktok-ads"
  });
};
var listFacebookPages = (options) => {
  return (options?.client ?? client).get({
    ...options,
    url: "/v1/connect/facebook/select-page"
  });
};
var selectFacebookPage = (options) => {
  return (options?.client ?? client).post({
    ...options,
    url: "/v1/connect/facebook/select-page"
  });
};
var listGoogleBusinessLocations = (options) => {
  return (options?.client ?? client).get({
    ...options,
    url: "/v1/connect/googlebusiness/locations"
  });
};
var selectGoogleBusinessLocation = (options) => {
  return (options?.client ?? client).post({
    ...options,
    url: "/v1/connect/googlebusiness/select-location"
  });
};
var getGoogleBusinessReviews = (options) => {
  return (options?.client ?? client).get({
    ...options,
    url: "/v1/accounts/{accountId}/gmb-reviews"
  });
};
var getGoogleBusinessVerifications = (options) => {
  return (options?.client ?? client).get({
    ...options,
    url: "/v1/accounts/{accountId}/gmb-verifications"
  });
};
var startGoogleBusinessVerification = (options) => {
  return (options?.client ?? client).post({
    ...options,
    url: "/v1/accounts/{accountId}/gmb-verifications"
  });
};
var fetchGoogleBusinessVerificationOptions = (options) => {
  return (options?.client ?? client).post({
    ...options,
    url: "/v1/accounts/{accountId}/gmb-verifications/options"
  });
};
var completeGoogleBusinessVerification = (options) => {
  return (options?.client ?? client).post({
    ...options,
    url: "/v1/accounts/{accountId}/gmb-verifications/{verificationId}/complete"
  });
};
var getGoogleBusinessFoodMenus = (options) => {
  return (options?.client ?? client).get({
    ...options,
    url: "/v1/accounts/{accountId}/gmb-food-menus"
  });
};
var updateGoogleBusinessFoodMenus = (options) => {
  return (options?.client ?? client).put({
    ...options,
    url: "/v1/accounts/{accountId}/gmb-food-menus"
  });
};
var getGoogleBusinessLocationDetails = (options) => {
  return (options?.client ?? client).get({
    ...options,
    url: "/v1/accounts/{accountId}/gmb-location-details"
  });
};
var updateGoogleBusinessLocationDetails = (options) => {
  return (options?.client ?? client).put({
    ...options,
    url: "/v1/accounts/{accountId}/gmb-location-details"
  });
};
var listGoogleBusinessMedia = (options) => {
  return (options?.client ?? client).get({
    ...options,
    url: "/v1/accounts/{accountId}/gmb-media"
  });
};
var createGoogleBusinessMedia = (options) => {
  return (options?.client ?? client).post({
    ...options,
    url: "/v1/accounts/{accountId}/gmb-media"
  });
};
var deleteGoogleBusinessMedia = (options) => {
  return (options?.client ?? client).delete({
    ...options,
    url: "/v1/accounts/{accountId}/gmb-media"
  });
};
var getGmbAttributeMetadata = (options) => {
  return (options?.client ?? client).get({
    ...options,
    url: "/v1/accounts/{accountId}/gmb-attribute-metadata"
  });
};
var getGoogleBusinessAttributes = (options) => {
  return (options?.client ?? client).get({
    ...options,
    url: "/v1/accounts/{accountId}/gmb-attributes"
  });
};
var updateGoogleBusinessAttributes = (options) => {
  return (options?.client ?? client).put({
    ...options,
    url: "/v1/accounts/{accountId}/gmb-attributes"
  });
};
var listGoogleBusinessPlaceActions = (options) => {
  return (options?.client ?? client).get({
    ...options,
    url: "/v1/accounts/{accountId}/gmb-place-actions"
  });
};
var createGoogleBusinessPlaceAction = (options) => {
  return (options?.client ?? client).post({
    ...options,
    url: "/v1/accounts/{accountId}/gmb-place-actions"
  });
};
var deleteGoogleBusinessPlaceAction = (options) => {
  return (options?.client ?? client).delete({
    ...options,
    url: "/v1/accounts/{accountId}/gmb-place-actions"
  });
};
var updateGoogleBusinessPlaceAction = (options) => {
  return (options?.client ?? client).patch({
    ...options,
    url: "/v1/accounts/{accountId}/gmb-place-actions"
  });
};
var getGoogleBusinessServices = (options) => {
  return (options?.client ?? client).get({
    ...options,
    url: "/v1/accounts/{accountId}/gmb-services"
  });
};
var updateGoogleBusinessServices = (options) => {
  return (options?.client ?? client).put({
    ...options,
    url: "/v1/accounts/{accountId}/gmb-services"
  });
};
var batchGetGoogleBusinessReviews = (options) => {
  return (options?.client ?? client).post({
    ...options,
    url: "/v1/accounts/{accountId}/gmb-reviews/batch"
  });
};
var replyToGoogleBusinessReview = (options) => {
  return (options?.client ?? client).post({
    ...options,
    url: "/v1/accounts/{accountId}/gmb-reviews/{reviewId}/reply"
  });
};
var deleteGoogleBusinessReviewReply = (options) => {
  return (options?.client ?? client).delete({
    ...options,
    url: "/v1/accounts/{accountId}/gmb-reviews/{reviewId}/reply"
  });
};
var getPendingOAuthData = (options) => {
  return (options?.client ?? client).get({
    ...options,
    url: "/v1/connect/pending-data"
  });
};
var listLinkedInOrganizations = (options) => {
  return (options?.client ?? client).get({
    ...options,
    url: "/v1/connect/linkedin/organizations"
  });
};
var selectLinkedInOrganization = (options) => {
  return (options?.client ?? client).post({
    ...options,
    url: "/v1/connect/linkedin/select-organization"
  });
};
var listPinterestBoardsForSelection = (options) => {
  return (options?.client ?? client).get({
    ...options,
    url: "/v1/connect/pinterest/select-board"
  });
};
var selectPinterestBoard = (options) => {
  return (options?.client ?? client).post({
    ...options,
    url: "/v1/connect/pinterest/select-board"
  });
};
var listSnapchatProfiles = (options) => {
  return (options?.client ?? client).get({
    ...options,
    url: "/v1/connect/snapchat/select-profile"
  });
};
var selectSnapchatProfile = (options) => {
  return (options?.client ?? client).post({
    ...options,
    url: "/v1/connect/snapchat/select-profile"
  });
};
var connectBlueskyCredentials = (options) => {
  return (options?.client ?? client).post({
    ...options,
    url: "/v1/connect/bluesky/credentials"
  });
};
var connectWhatsAppCredentials = (options) => {
  return (options?.client ?? client).post({
    ...options,
    url: "/v1/connect/whatsapp/credentials"
  });
};
var listWhatsAppPhoneNumbers = (options) => {
  return (options?.client ?? client).get({
    ...options,
    url: "/v1/connect/whatsapp/select-phone-number"
  });
};
var completeWhatsAppPhoneSelection = (options) => {
  return (options?.client ?? client).post({
    ...options,
    url: "/v1/connect/whatsapp/select-phone-number"
  });
};
var getTelegramConnectStatus = (options) => {
  return (options?.client ?? client).get({
    ...options,
    url: "/v1/connect/telegram"
  });
};
var initiateTelegramConnect = (options) => {
  return (options?.client ?? client).post({
    ...options,
    url: "/v1/connect/telegram"
  });
};
var completeTelegramConnect = (options) => {
  return (options?.client ?? client).patch({
    ...options,
    url: "/v1/connect/telegram"
  });
};
var getFacebookPages = (options) => {
  return (options?.client ?? client).get({
    ...options,
    url: "/v1/accounts/{accountId}/facebook-page"
  });
};
var updateFacebookPage = (options) => {
  return (options?.client ?? client).put({
    ...options,
    url: "/v1/accounts/{accountId}/facebook-page"
  });
};
var getLinkedInOrganizations = (options) => {
  return (options?.client ?? client).get({
    ...options,
    url: "/v1/accounts/{accountId}/linkedin-organizations"
  });
};
var getLinkedInAggregateAnalytics = (options) => {
  return (options?.client ?? client).get({
    ...options,
    url: "/v1/accounts/{accountId}/linkedin-aggregate-analytics"
  });
};
var getLinkedInPostAnalytics = (options) => {
  return (options?.client ?? client).get({
    ...options,
    url: "/v1/accounts/{accountId}/linkedin-post-analytics"
  });
};
var getLinkedInPostReactions = (options) => {
  return (options?.client ?? client).get({
    ...options,
    url: "/v1/accounts/{accountId}/linkedin-post-reactions"
  });
};
var updateLinkedInOrganization = (options) => {
  return (options?.client ?? client).put({
    ...options,
    url: "/v1/accounts/{accountId}/linkedin-organization"
  });
};
var getLinkedInMentions = (options) => {
  return (options?.client ?? client).get({
    ...options,
    url: "/v1/accounts/{accountId}/linkedin-mentions"
  });
};
var listInstagramStories = (options) => {
  return (options?.client ?? client).get({
    ...options,
    url: "/v1/accounts/{accountId}/instagram/stories"
  });
};
var getInstagramPublishingLimit = (options) => {
  return (options?.client ?? client).get({
    ...options,
    url: "/v1/accounts/{accountId}/instagram/publishing-limit"
  });
};
var getInstagramStoryInsights = (options) => {
  return (options?.client ?? client).get({
    ...options,
    url: "/v1/accounts/{accountId}/instagram/stories/{storyId}/insights"
  });
};
var getPinterestBoards = (options) => {
  return (options?.client ?? client).get({
    ...options,
    url: "/v1/accounts/{accountId}/pinterest-boards"
  });
};
var updatePinterestBoards = (options) => {
  return (options?.client ?? client).put({
    ...options,
    url: "/v1/accounts/{accountId}/pinterest-boards"
  });
};
var createPinterestBoard = (options) => {
  return (options?.client ?? client).post({
    ...options,
    url: "/v1/accounts/{accountId}/pinterest-boards"
  });
};
var getYoutubePlaylists = (options) => {
  return (options?.client ?? client).get({
    ...options,
    url: "/v1/accounts/{accountId}/youtube-playlists"
  });
};
var updateYoutubeDefaultPlaylist = (options) => {
  return (options?.client ?? client).put({
    ...options,
    url: "/v1/accounts/{accountId}/youtube-playlists"
  });
};
var getGmbLocations = (options) => {
  return (options?.client ?? client).get({
    ...options,
    url: "/v1/accounts/{accountId}/gmb-locations"
  });
};
var updateGmbLocation = (options) => {
  return (options?.client ?? client).put({
    ...options,
    url: "/v1/accounts/{accountId}/gmb-locations"
  });
};
var getFacebookPostReactions = (options) => {
  return (options?.client ?? client).get({
    ...options,
    url: "/v1/accounts/{accountId}/facebook-post-reactions"
  });
};
var getRedditSubreddits = (options) => {
  return (options?.client ?? client).get({
    ...options,
    url: "/v1/accounts/{accountId}/reddit-subreddits"
  });
};
var updateRedditSubreddits = (options) => {
  return (options?.client ?? client).put({
    ...options,
    url: "/v1/accounts/{accountId}/reddit-subreddits"
  });
};
var getSubredditRules = (options) => {
  return (options?.client ?? client).get({
    ...options,
    url: "/v1/accounts/{accountId}/reddit-subreddits/{subreddit}/rules"
  });
};
var voteRedditThing = (options) => {
  return (options?.client ?? client).post({
    ...options,
    url: "/v1/accounts/{accountId}/reddit-vote"
  });
};
var getRedditFlairs = (options) => {
  return (options?.client ?? client).get({
    ...options,
    url: "/v1/accounts/{accountId}/reddit-flairs"
  });
};
var setRedditPostFlair = (options) => {
  return (options?.client ?? client).post({
    ...options,
    url: "/v1/accounts/{accountId}/reddit-flairs"
  });
};
var getDiscordSettings = (options) => {
  return (options?.client ?? client).get({
    ...options,
    url: "/v1/accounts/{accountId}/discord-settings"
  });
};
var updateDiscordSettings = (options) => {
  return (options?.client ?? client).patch({
    ...options,
    url: "/v1/accounts/{accountId}/discord-settings"
  });
};
var getDiscordChannels = (options) => {
  return (options?.client ?? client).get({
    ...options,
    url: "/v1/accounts/{accountId}/discord-channels"
  });
};
var sendDiscordDirectMessage = (options) => {
  return (options?.client ?? client).post({
    ...options,
    url: "/v1/discord/dms"
  });
};
var listDiscordGuildRoles = (options) => {
  return (options?.client ?? client).get({
    ...options,
    url: "/v1/discord/guilds/{guildId}/roles"
  });
};
var createDiscordGuildRole = (options) => {
  return (options?.client ?? client).post({
    ...options,
    url: "/v1/discord/guilds/{guildId}/roles"
  });
};
var editDiscordGuildRole = (options) => {
  return (options?.client ?? client).patch({
    ...options,
    url: "/v1/discord/guilds/{guildId}/roles/{roleId}"
  });
};
var deleteDiscordGuildRole = (options) => {
  return (options?.client ?? client).delete({
    ...options,
    url: "/v1/discord/guilds/{guildId}/roles/{roleId}"
  });
};
var listDiscordGuildMembers = (options) => {
  return (options?.client ?? client).get({
    ...options,
    url: "/v1/discord/guilds/{guildId}/members"
  });
};
var addDiscordMemberRole = (options) => {
  return (options?.client ?? client).put({
    ...options,
    url: "/v1/discord/guilds/{guildId}/members/{userId}/roles/{roleId}"
  });
};
var removeDiscordMemberRole = (options) => {
  return (options?.client ?? client).delete({
    ...options,
    url: "/v1/discord/guilds/{guildId}/members/{userId}/roles/{roleId}"
  });
};
var deleteDiscordMessage = (options) => {
  return (options?.client ?? client).delete({
    ...options,
    url: "/v1/discord/channels/{channelId}/messages/{messageId}"
  });
};
var crosspostDiscordMessage = (options) => {
  return (options?.client ?? client).post({
    ...options,
    url: "/v1/discord/channels/{channelId}/messages/{messageId}/crosspost"
  });
};
var createDiscordThread = (options) => {
  return (options?.client ?? client).post({
    ...options,
    url: "/v1/discord/channels/{channelId}/threads"
  });
};
var listDiscordPinnedMessages = (options) => {
  return (options?.client ?? client).get({
    ...options,
    url: "/v1/discord/channels/{channelId}/pins"
  });
};
var pinDiscordMessage = (options) => {
  return (options?.client ?? client).put({
    ...options,
    url: "/v1/discord/channels/{channelId}/pins/{messageId}"
  });
};
var unpinDiscordMessage = (options) => {
  return (options?.client ?? client).delete({
    ...options,
    url: "/v1/discord/channels/{channelId}/pins/{messageId}"
  });
};
var listDiscordScheduledEvents = (options) => {
  return (options?.client ?? client).get({
    ...options,
    url: "/v1/discord/guilds/{guildId}/events"
  });
};
var createDiscordScheduledEvent = (options) => {
  return (options?.client ?? client).post({
    ...options,
    url: "/v1/discord/guilds/{guildId}/events"
  });
};
var getDiscordScheduledEvent = (options) => {
  return (options?.client ?? client).get({
    ...options,
    url: "/v1/discord/guilds/{guildId}/events/{eventId}"
  });
};
var updateDiscordScheduledEvent = (options) => {
  return (options?.client ?? client).patch({
    ...options,
    url: "/v1/discord/guilds/{guildId}/events/{eventId}"
  });
};
var deleteDiscordScheduledEvent = (options) => {
  return (options?.client ?? client).delete({
    ...options,
    url: "/v1/discord/guilds/{guildId}/events/{eventId}"
  });
};
var listQueueSlots = (options) => {
  return (options?.client ?? client).get({
    ...options,
    url: "/v1/queue/slots"
  });
};
var createQueueSlot = (options) => {
  return (options?.client ?? client).post({
    ...options,
    url: "/v1/queue/slots"
  });
};
var updateQueueSlot = (options) => {
  return (options?.client ?? client).put({
    ...options,
    url: "/v1/queue/slots"
  });
};
var deleteQueueSlot = (options) => {
  return (options?.client ?? client).delete({
    ...options,
    url: "/v1/queue/slots"
  });
};
var previewQueue = (options) => {
  return (options?.client ?? client).get({
    ...options,
    url: "/v1/queue/preview"
  });
};
var getNextQueueSlot = (options) => {
  return (options?.client ?? client).get({
    ...options,
    url: "/v1/queue/next-slot"
  });
};
var getWebhookSettings = (options) => {
  return (options?.client ?? client).get({
    ...options,
    url: "/v1/webhooks/settings"
  });
};
var createWebhookSettings = (options) => {
  return (options?.client ?? client).post({
    ...options,
    url: "/v1/webhooks/settings"
  });
};
var updateWebhookSettings = (options) => {
  return (options?.client ?? client).put({
    ...options,
    url: "/v1/webhooks/settings"
  });
};
var deleteWebhookSettings = (options) => {
  return (options?.client ?? client).delete({
    ...options,
    url: "/v1/webhooks/settings"
  });
};
var getWebhookLogs = (options) => {
  return (options?.client ?? client).get({
    ...options,
    url: "/v1/webhooks/logs"
  });
};
var testWebhook = (options) => {
  return (options?.client ?? client).post({
    ...options,
    url: "/v1/webhooks/test"
  });
};
var listLogs = (options) => {
  return (options?.client ?? client).get({
    ...options,
    url: "/v1/logs"
  });
};
var listInboxConversations = (options) => {
  return (options?.client ?? client).get({
    ...options,
    url: "/v1/inbox/conversations"
  });
};
var createInboxConversation = (options) => {
  return (options?.client ?? client).post({
    ...options,
    url: "/v1/inbox/conversations"
  });
};
var searchInboxConversations = (options) => {
  return (options?.client ?? client).get({
    ...options,
    url: "/v1/inbox/conversations/search"
  });
};
var getInboxConversation = (options) => {
  return (options?.client ?? client).get({
    ...options,
    url: "/v1/inbox/conversations/{conversationId}"
  });
};
var updateInboxConversation = (options) => {
  return (options?.client ?? client).put({
    ...options,
    url: "/v1/inbox/conversations/{conversationId}"
  });
};
var getInboxConversationMessages = (options) => {
  return (options?.client ?? client).get({
    ...options,
    url: "/v1/inbox/conversations/{conversationId}/messages"
  });
};
var sendInboxMessage = (options) => {
  return (options?.client ?? client).post({
    ...options,
    url: "/v1/inbox/conversations/{conversationId}/messages"
  });
};
var editInboxMessage = (options) => {
  return (options?.client ?? client).patch({
    ...options,
    url: "/v1/inbox/conversations/{conversationId}/messages/{messageId}"
  });
};
var deleteInboxMessage = (options) => {
  return (options?.client ?? client).delete({
    ...options,
    url: "/v1/inbox/conversations/{conversationId}/messages/{messageId}"
  });
};
var sendTypingIndicator = (options) => {
  return (options?.client ?? client).post({
    ...options,
    url: "/v1/inbox/conversations/{conversationId}/typing"
  });
};
var markConversationRead = (options) => {
  return (options?.client ?? client).post({
    ...options,
    url: "/v1/inbox/conversations/{conversationId}/read"
  });
};
var addMessageReaction = (options) => {
  return (options?.client ?? client).post({
    ...options,
    url: "/v1/inbox/conversations/{conversationId}/messages/{messageId}/reactions"
  });
};
var removeMessageReaction = (options) => {
  return (options?.client ?? client).delete({
    ...options,
    url: "/v1/inbox/conversations/{conversationId}/messages/{messageId}/reactions"
  });
};
var uploadMediaDirect = (options) => {
  return (options?.client ?? client).post({
    ...options,
    ...W,
    headers: {
      "Content-Type": null,
      ...options?.headers
    },
    url: "/v1/media/upload-direct"
  });
};
var getMessengerMenu = (options) => {
  return (options?.client ?? client).get({
    ...options,
    url: "/v1/accounts/{accountId}/messenger-menu"
  });
};
var setMessengerMenu = (options) => {
  return (options?.client ?? client).put({
    ...options,
    url: "/v1/accounts/{accountId}/messenger-menu"
  });
};
var deleteMessengerMenu = (options) => {
  return (options?.client ?? client).delete({
    ...options,
    url: "/v1/accounts/{accountId}/messenger-menu"
  });
};
var getInstagramIceBreakers = (options) => {
  return (options?.client ?? client).get({
    ...options,
    url: "/v1/accounts/{accountId}/instagram-ice-breakers"
  });
};
var setInstagramIceBreakers = (options) => {
  return (options?.client ?? client).put({
    ...options,
    url: "/v1/accounts/{accountId}/instagram-ice-breakers"
  });
};
var deleteInstagramIceBreakers = (options) => {
  return (options?.client ?? client).delete({
    ...options,
    url: "/v1/accounts/{accountId}/instagram-ice-breakers"
  });
};
var getTelegramCommands = (options) => {
  return (options?.client ?? client).get({
    ...options,
    url: "/v1/accounts/{accountId}/telegram-commands"
  });
};
var setTelegramCommands = (options) => {
  return (options?.client ?? client).put({
    ...options,
    url: "/v1/accounts/{accountId}/telegram-commands"
  });
};
var deleteTelegramCommands = (options) => {
  return (options?.client ?? client).delete({
    ...options,
    url: "/v1/accounts/{accountId}/telegram-commands"
  });
};
var listInboxComments = (options) => {
  return (options?.client ?? client).get({
    ...options,
    url: "/v1/inbox/comments"
  });
};
var getInboxPostComments = (options) => {
  return (options?.client ?? client).get({
    ...options,
    url: "/v1/inbox/comments/{postId}"
  });
};
var replyToInboxPost = (options) => {
  return (options?.client ?? client).post({
    ...options,
    url: "/v1/inbox/comments/{postId}"
  });
};
var deleteInboxComment = (options) => {
  return (options?.client ?? client).delete({
    ...options,
    url: "/v1/inbox/comments/{postId}"
  });
};
var editInboxComment = (options) => {
  return (options?.client ?? client).patch({
    ...options,
    url: "/v1/inbox/comments/{postId}/{commentId}"
  });
};
var setCommentModeration = (options) => {
  return (options?.client ?? client).post({
    ...options,
    url: "/v1/inbox/comments/{postId}/{commentId}/moderation"
  });
};
var hideInboxComment = (options) => {
  return (options?.client ?? client).post({
    ...options,
    url: "/v1/inbox/comments/{postId}/{commentId}/hide"
  });
};
var unhideInboxComment = (options) => {
  return (options?.client ?? client).delete({
    ...options,
    url: "/v1/inbox/comments/{postId}/{commentId}/hide"
  });
};
var likeInboxComment = (options) => {
  return (options?.client ?? client).post({
    ...options,
    url: "/v1/inbox/comments/{postId}/{commentId}/like"
  });
};
var unlikeInboxComment = (options) => {
  return (options?.client ?? client).delete({
    ...options,
    url: "/v1/inbox/comments/{postId}/{commentId}/like"
  });
};
var sendPrivateReplyToComment = (options) => {
  return (options?.client ?? client).post({
    ...options,
    url: "/v1/inbox/comments/{postId}/{commentId}/private-reply"
  });
};
var retweetPost = (options) => {
  return (options?.client ?? client).post({
    ...options,
    url: "/v1/twitter/retweet"
  });
};
var undoRetweet = (options) => {
  return (options?.client ?? client).delete({
    ...options,
    url: "/v1/twitter/retweet"
  });
};
var bookmarkPost = (options) => {
  return (options?.client ?? client).post({
    ...options,
    url: "/v1/twitter/bookmark"
  });
};
var removeBookmark = (options) => {
  return (options?.client ?? client).delete({
    ...options,
    url: "/v1/twitter/bookmark"
  });
};
var followUser = (options) => {
  return (options?.client ?? client).post({
    ...options,
    url: "/v1/twitter/follow"
  });
};
var unfollowUser = (options) => {
  return (options?.client ?? client).delete({
    ...options,
    url: "/v1/twitter/follow"
  });
};
var listInboxMentions = (options) => {
  return (options?.client ?? client).get({
    ...options,
    url: "/v1/inbox/mentions"
  });
};
var replyToMention = (options) => {
  return (options?.client ?? client).post({
    ...options,
    url: "/v1/inbox/mentions/reply"
  });
};
var listInboxReviews = (options) => {
  return (options?.client ?? client).get({
    ...options,
    url: "/v1/inbox/reviews"
  });
};
var replyToInboxReview = (options) => {
  return (options?.client ?? client).post({
    ...options,
    url: "/v1/inbox/reviews/{reviewId}/reply"
  });
};
var deleteInboxReviewReply = (options) => {
  return (options?.client ?? client).delete({
    ...options,
    url: "/v1/inbox/reviews/{reviewId}/reply"
  });
};
var getWhatsAppTemplates = (options) => {
  return (options?.client ?? client).get({
    ...options,
    url: "/v1/whatsapp/templates"
  });
};
var createWhatsAppTemplate = (options) => {
  return (options?.client ?? client).post({
    ...options,
    url: "/v1/whatsapp/templates"
  });
};
var getWhatsAppTemplate = (options) => {
  return (options?.client ?? client).get({
    ...options,
    url: "/v1/whatsapp/templates/{templateName}"
  });
};
var updateWhatsAppTemplate = (options) => {
  return (options?.client ?? client).patch({
    ...options,
    url: "/v1/whatsapp/templates/{templateName}"
  });
};
var deleteWhatsAppTemplate = (options) => {
  return (options?.client ?? client).delete({
    ...options,
    url: "/v1/whatsapp/templates/{templateName}"
  });
};
var getWhatsAppCallingConfig = (options) => {
  return (options?.client ?? client).get({
    ...options,
    url: "/v1/whatsapp/calling"
  });
};
var enableWhatsAppCallingLegacy = (options) => {
  return (options?.client ?? client).post({
    ...options,
    url: "/v1/whatsapp/phone-numbers/{id}/calling"
  });
};
var updateWhatsAppCallingLegacy = (options) => {
  return (options?.client ?? client).patch({
    ...options,
    url: "/v1/whatsapp/phone-numbers/{id}/calling"
  });
};
var disableWhatsAppCallingLegacy = (options) => {
  return (options?.client ?? client).delete({
    ...options,
    url: "/v1/whatsapp/phone-numbers/{id}/calling"
  });
};
var getWhatsAppCallPermissions = (options) => {
  return (options?.client ?? client).get({
    ...options,
    url: "/v1/whatsapp/call-permissions"
  });
};
var initiateWhatsAppCall = (options) => {
  return (options?.client ?? client).post({
    ...options,
    url: "/v1/whatsapp/calls"
  });
};
var listWhatsAppCalls = (options) => {
  return (options?.client ?? client).get({
    ...options,
    url: "/v1/whatsapp/calls"
  });
};
var getWhatsAppCall = (options) => {
  return (options?.client ?? client).get({
    ...options,
    url: "/v1/whatsapp/calls/{id}"
  });
};
var getWhatsAppCallRecording = (options) => {
  return (options?.client ?? client).get({
    ...options,
    url: "/v1/whatsapp/calls/{id}/recording"
  });
};
var getWhatsAppCallEstimate = (options) => {
  return (options?.client ?? client).get({
    ...options,
    url: "/v1/whatsapp/calls/estimate"
  });
};
var listCalls = (options) => {
  return (options?.client ?? client).get({
    ...options,
    url: "/v1/calls"
  });
};
var getCall = (options) => {
  return (options?.client ?? client).get({
    ...options,
    url: "/v1/calls/{id}"
  });
};
var getCallRecording = (options) => {
  return (options?.client ?? client).get({
    ...options,
    url: "/v1/calls/{id}/recording"
  });
};
var createVoiceCall = (options) => {
  return (options?.client ?? client).post({
    ...options,
    url: "/v1/voice/calls"
  });
};
var listVoiceCalls = (options) => {
  return (options?.client ?? client).get({
    ...options,
    url: "/v1/voice/calls"
  });
};
var getVoiceCall = (options) => {
  return (options?.client ?? client).get({
    ...options,
    url: "/v1/voice/calls/{id}"
  });
};
var endVoiceCall = (options) => {
  return (options?.client ?? client).post({
    ...options,
    url: "/v1/voice/calls/{id}/end"
  });
};
var getVoiceCallRecording = (options) => {
  return (options?.client ?? client).get({
    ...options,
    url: "/v1/voice/calls/{id}/recording"
  });
};
var transferVoiceCall = (options) => {
  return (options?.client ?? client).post({
    ...options,
    url: "/v1/voice/calls/{id}/transfer"
  });
};
var getVoiceCallEstimate = (options) => {
  return (options?.client ?? client).get({
    ...options,
    url: "/v1/voice/calls/estimate"
  });
};
var createVoiceWebSession = (options) => {
  return (options?.client ?? client).post({
    ...options,
    url: "/v1/voice/calls/web"
  });
};
var dialVoiceWebCall = (options) => {
  return (options?.client ?? client).post({
    ...options,
    url: "/v1/voice/calls/web/dial"
  });
};
var sendSms = (options) => {
  return (options?.client ?? client).post({
    ...options,
    url: "/v1/sms/messages"
  });
};
var lookupSmsNumber = (options) => {
  return (options?.client ?? client).get({
    ...options,
    url: "/v1/sms/lookup"
  });
};
var listSmsOptOuts = (options) => {
  return (options?.client ?? client).get({
    ...options,
    url: "/v1/sms/opt-outs"
  });
};
var startSmsRegistration = (options) => {
  return (options?.client ?? client).post({
    ...options,
    url: "/v1/sms/registrations"
  });
};
var listSmsRegistrations = (options) => {
  return (options?.client ?? client).get({
    ...options,
    url: "/v1/sms/registrations"
  });
};
var deactivateSmsRegistration = (options) => {
  return (options?.client ?? client).delete({
    ...options,
    url: "/v1/sms/registrations/{id}"
  });
};
var getSmsRegistration = (options) => {
  return (options?.client ?? client).get({
    ...options,
    url: "/v1/sms/registrations/{id}"
  });
};
var verifySmsRegistrationOtp = (options) => {
  return (options?.client ?? client).post({
    ...options,
    url: "/v1/sms/registrations/{id}/verify-otp"
  });
};
var resendSmsRegistrationOtp = (options) => {
  return (options?.client ?? client).post({
    ...options,
    url: "/v1/sms/registrations/{id}/resend-otp"
  });
};
var appealSmsRegistration = (options) => {
  return (options?.client ?? client).post({
    ...options,
    url: "/v1/sms/registrations/{id}/appeal"
  });
};
var uploadSmsOptInProofFile = (options) => {
  return (options?.client ?? client).post({
    ...options,
    ...W,
    headers: {
      "Content-Type": null,
      ...options?.headers
    },
    url: "/v1/sms/opt-in-proof"
  });
};
var uploadSmsOptInProof = (options) => {
  return (options?.client ?? client).post({
    ...options,
    ...W,
    headers: {
      "Content-Type": null,
      ...options?.headers
    },
    url: "/v1/sms/registrations/{id}/opt-in-proof"
  });
};
var shareSmsRegistration = (options) => {
  return (options?.client ?? client).post({
    ...options,
    url: "/v1/sms/registrations/share"
  });
};
var getWhatsAppLibraryTemplate = (options) => {
  return (options?.client ?? client).get({
    ...options,
    url: "/v1/whatsapp/template-library"
  });
};
var getWhatsAppBusinessProfile = (options) => {
  return (options?.client ?? client).get({
    ...options,
    url: "/v1/whatsapp/business-profile"
  });
};
var updateWhatsAppBusinessProfile = (options) => {
  return (options?.client ?? client).post({
    ...options,
    url: "/v1/whatsapp/business-profile"
  });
};
var uploadWhatsAppProfilePhoto = (options) => {
  return (options?.client ?? client).post({
    ...options,
    ...W,
    headers: {
      "Content-Type": null,
      ...options?.headers
    },
    url: "/v1/whatsapp/business-profile/photo"
  });
};
var getWhatsAppDisplayName = (options) => {
  return (options?.client ?? client).get({
    ...options,
    url: "/v1/whatsapp/business-profile/display-name"
  });
};
var updateWhatsAppDisplayName = (options) => {
  return (options?.client ?? client).post({
    ...options,
    url: "/v1/whatsapp/business-profile/display-name"
  });
};
var getWhatsappBusinessUsername = (options) => {
  return (options?.client ?? client).get({
    ...options,
    url: "/v1/whatsapp/business-profile/username"
  });
};
var setWhatsappBusinessUsername = (options) => {
  return (options?.client ?? client).post({
    ...options,
    url: "/v1/whatsapp/business-profile/username"
  });
};
var deleteWhatsappBusinessUsername = (options) => {
  return (options?.client ?? client).delete({
    ...options,
    url: "/v1/whatsapp/business-profile/username"
  });
};
var getWhatsappBusinessUsernameSuggestions = (options) => {
  return (options?.client ?? client).get({
    ...options,
    url: "/v1/whatsapp/business-profile/username/suggestions"
  });
};
var getWhatsAppNumberInfo = (options) => {
  return (options?.client ?? client).get({
    ...options,
    url: "/v1/whatsapp/number-info"
  });
};
var getWhatsAppBlockStatus = (options) => {
  return (options?.client ?? client).get({
    ...options,
    url: "/v1/whatsapp/block-users/status"
  });
};
var getWhatsAppBlockedUsers = (options) => {
  return (options?.client ?? client).get({
    ...options,
    url: "/v1/whatsapp/block-users"
  });
};
var blockWhatsAppUsers = (options) => {
  return (options?.client ?? client).post({
    ...options,
    url: "/v1/whatsapp/block-users"
  });
};
var unblockWhatsAppUsers = (options) => {
  return (options?.client ?? client).delete({
    ...options,
    url: "/v1/whatsapp/block-users"
  });
};
var getWhatsAppDataset = (options) => {
  return (options?.client ?? client).get({
    ...options,
    url: "/v1/whatsapp/dataset"
  });
};
var createWhatsAppDataset = (options) => {
  return (options?.client ?? client).post({
    ...options,
    url: "/v1/whatsapp/dataset"
  });
};
var listPhoneNumbers = (options) => {
  return (options?.client ?? client).get({
    ...options,
    url: "/v1/phone-numbers"
  });
};
var getPhoneNumber = (options) => {
  return (options?.client ?? client).get({
    ...options,
    url: "/v1/phone-numbers/{id}"
  });
};
var releasePhoneNumber = (options) => {
  return (options?.client ?? client).delete({
    ...options,
    url: "/v1/phone-numbers/{id}"
  });
};
var purchasePhoneNumber = (options) => {
  return (options?.client ?? client).post({
    ...options,
    url: "/v1/phone-numbers/purchase"
  });
};
var listPhoneNumberCountries = (options) => {
  return (options?.client ?? client).get({
    ...options,
    url: "/v1/phone-numbers/countries"
  });
};
var searchAvailablePhoneNumbers = (options) => {
  return (options?.client ?? client).get({
    ...options,
    url: "/v1/phone-numbers/available"
  });
};
var checkPhoneNumberAvailability = (options) => {
  return (options?.client ?? client).get({
    ...options,
    url: "/v1/phone-numbers/availability"
  });
};
var getWhatsAppPhoneNumbers = (options) => {
  return (options?.client ?? client).get({
    ...options,
    url: "/v1/whatsapp/phone-numbers"
  });
};
var purchaseWhatsAppPhoneNumber = (options) => {
  return (options?.client ?? client).post({
    ...options,
    url: "/v1/whatsapp/phone-numbers/purchase"
  });
};
var listWhatsAppNumberCountries = (options) => {
  return (options?.client ?? client).get({
    ...options,
    url: "/v1/whatsapp/phone-numbers/countries"
  });
};
var searchAvailableWhatsAppNumbers = (options) => {
  return (options?.client ?? client).get({
    ...options,
    url: "/v1/whatsapp/phone-numbers/available"
  });
};
var checkWhatsAppNumberAvailability = (options) => {
  return (options?.client ?? client).get({
    ...options,
    url: "/v1/whatsapp/phone-numbers/availability"
  });
};
var getPhoneNumberKycForm = (options) => {
  return (options?.client ?? client).get({
    ...options,
    url: "/v1/phone-numbers/kyc"
  });
};
var submitPhoneNumberKyc = (options) => {
  return (options?.client ?? client).post({
    ...options,
    url: "/v1/phone-numbers/kyc"
  });
};
var uploadPhoneNumberKycDocument = (options) => {
  return (options?.client ?? client).post({
    ...options,
    url: "/v1/phone-numbers/kyc/upload-document"
  });
};
var validatePhoneNumberKycAddress = (options) => {
  return (options?.client ?? client).post({
    ...options,
    url: "/v1/phone-numbers/kyc/validate-address"
  });
};
var createPhoneNumberKycLink = (options) => {
  return (options?.client ?? client).post({
    ...options,
    url: "/v1/phone-numbers/kyc/share"
  });
};
var createPhoneNumberPortIn = (options) => {
  return (options?.client ?? client).post({
    ...options,
    url: "/v1/phone-numbers/port-in"
  });
};
var listPhoneNumberPortIns = (options) => {
  return (options?.client ?? client).get({
    ...options,
    url: "/v1/phone-numbers/port-in"
  });
};
var checkPhoneNumberPortability = (options) => {
  return (options?.client ?? client).post({
    ...options,
    url: "/v1/phone-numbers/port-in/check"
  });
};
var uploadPhoneNumberPortInDocument = (options) => {
  return (options?.client ?? client).post({
    ...options,
    ...W,
    headers: {
      "Content-Type": null,
      ...options?.headers
    },
    url: "/v1/phone-numbers/port-in/documents"
  });
};
var cancelPhoneNumberPortIn = (options) => {
  return (options?.client ?? client).delete({
    ...options,
    url: "/v1/phone-numbers/port-in/{id}"
  });
};
var reviewPhoneNumberKycPacket = (options) => {
  return (options?.client ?? client).post({
    ...options,
    url: "/v1/phone-numbers/kyc/review-packet"
  });
};
var getPhoneNumberRemediation = (options) => {
  return (options?.client ?? client).get({
    ...options,
    url: "/v1/phone-numbers/{id}/remediate"
  });
};
var remediatePhoneNumber = (options) => {
  return (options?.client ?? client).post({
    ...options,
    url: "/v1/phone-numbers/{id}/remediate"
  });
};
var getWhatsAppNumberKycForm = (options) => {
  return (options?.client ?? client).get({
    ...options,
    url: "/v1/whatsapp/phone-numbers/kyc"
  });
};
var submitWhatsAppNumberKyc = (options) => {
  return (options?.client ?? client).post({
    ...options,
    url: "/v1/whatsapp/phone-numbers/kyc"
  });
};
var uploadWhatsAppNumberKycDocument = (options) => {
  return (options?.client ?? client).post({
    ...options,
    url: "/v1/whatsapp/phone-numbers/kyc/upload-document"
  });
};
var validateWhatsAppNumberKycAddress = (options) => {
  return (options?.client ?? client).post({
    ...options,
    url: "/v1/whatsapp/phone-numbers/kyc/validate-address"
  });
};
var createWhatsAppNumberKycLink = (options) => {
  return (options?.client ?? client).post({
    ...options,
    url: "/v1/whatsapp/phone-numbers/kyc/share"
  });
};
var getWhatsAppNumberRemediation = (options) => {
  return (options?.client ?? client).get({
    ...options,
    url: "/v1/whatsapp/phone-numbers/{id}/remediate"
  });
};
var remediateWhatsAppNumber = (options) => {
  return (options?.client ?? client).post({
    ...options,
    url: "/v1/whatsapp/phone-numbers/{id}/remediate"
  });
};
var enableVoiceOnNumber = (options) => {
  return (options?.client ?? client).post({
    ...options,
    url: "/v1/phone-numbers/{id}/voice"
  });
};
var disableVoiceOnNumber = (options) => {
  return (options?.client ?? client).delete({
    ...options,
    url: "/v1/phone-numbers/{id}/voice"
  });
};
var enableSmsOnNumber = (options) => {
  return (options?.client ?? client).post({
    ...options,
    url: "/v1/phone-numbers/{id}/sms"
  });
};
var disableSmsOnNumber = (options) => {
  return (options?.client ?? client).delete({
    ...options,
    url: "/v1/phone-numbers/{id}/sms"
  });
};
var reuseSmsRegistrationForNumber = (options) => {
  return (options?.client ?? client).post({
    ...options,
    url: "/v1/phone-numbers/{id}/sms/reuse-registration"
  });
};
var getWhatsAppCalling = (options) => {
  return (options?.client ?? client).get({
    ...options,
    url: "/v1/phone-numbers/{id}/whatsapp/calling"
  });
};
var enableWhatsAppCalling = (options) => {
  return (options?.client ?? client).post({
    ...options,
    url: "/v1/phone-numbers/{id}/whatsapp/calling"
  });
};
var updateWhatsAppCalling = (options) => {
  return (options?.client ?? client).patch({
    ...options,
    url: "/v1/phone-numbers/{id}/whatsapp/calling"
  });
};
var disableWhatsAppCalling = (options) => {
  return (options?.client ?? client).delete({
    ...options,
    url: "/v1/phone-numbers/{id}/whatsapp/calling"
  });
};
var getWhatsAppPhoneNumber = (options) => {
  return (options?.client ?? client).get({
    ...options,
    url: "/v1/whatsapp/phone-numbers/{phoneNumberId}"
  });
};
var releaseWhatsAppPhoneNumber = (options) => {
  return (options?.client ?? client).delete({
    ...options,
    url: "/v1/whatsapp/phone-numbers/{phoneNumberId}"
  });
};
var listWhatsAppSandboxSessions = (options) => {
  return (options?.client ?? client).get({
    ...options,
    url: "/v1/whatsapp/sandbox/sessions"
  });
};
var createWhatsAppSandboxSession = (options) => {
  return (options?.client ?? client).post({
    ...options,
    url: "/v1/whatsapp/sandbox/sessions"
  });
};
var deleteWhatsAppSandboxSession = (options) => {
  return (options?.client ?? client).delete({
    ...options,
    url: "/v1/whatsapp/sandbox/sessions/{sessionId}"
  });
};
var listWhatsAppGroupChats = (options) => {
  return (options?.client ?? client).get({
    ...options,
    url: "/v1/whatsapp/wa-groups"
  });
};
var createWhatsAppGroupChat = (options) => {
  return (options?.client ?? client).post({
    ...options,
    url: "/v1/whatsapp/wa-groups"
  });
};
var getWhatsAppGroupChat = (options) => {
  return (options?.client ?? client).get({
    ...options,
    url: "/v1/whatsapp/wa-groups/{groupId}"
  });
};
var updateWhatsAppGroupChat = (options) => {
  return (options?.client ?? client).post({
    ...options,
    url: "/v1/whatsapp/wa-groups/{groupId}"
  });
};
var deleteWhatsAppGroupChat = (options) => {
  return (options?.client ?? client).delete({
    ...options,
    url: "/v1/whatsapp/wa-groups/{groupId}"
  });
};
var addWhatsAppGroupParticipants = (options) => {
  return (options?.client ?? client).post({
    ...options,
    url: "/v1/whatsapp/wa-groups/{groupId}/participants"
  });
};
var removeWhatsAppGroupParticipants = (options) => {
  return (options?.client ?? client).delete({
    ...options,
    url: "/v1/whatsapp/wa-groups/{groupId}/participants"
  });
};
var createWhatsAppGroupInviteLink = (options) => {
  return (options?.client ?? client).post({
    ...options,
    url: "/v1/whatsapp/wa-groups/{groupId}/invite-link"
  });
};
var listWhatsAppGroupJoinRequests = (options) => {
  return (options?.client ?? client).get({
    ...options,
    url: "/v1/whatsapp/wa-groups/{groupId}/join-requests"
  });
};
var approveWhatsAppGroupJoinRequests = (options) => {
  return (options?.client ?? client).post({
    ...options,
    url: "/v1/whatsapp/wa-groups/{groupId}/join-requests"
  });
};
var rejectWhatsAppGroupJoinRequests = (options) => {
  return (options?.client ?? client).delete({
    ...options,
    url: "/v1/whatsapp/wa-groups/{groupId}/join-requests"
  });
};
var listWhatsAppFlows = (options) => {
  return (options?.client ?? client).get({
    ...options,
    url: "/v1/whatsapp/flows"
  });
};
var createWhatsAppFlow = (options) => {
  return (options?.client ?? client).post({
    ...options,
    url: "/v1/whatsapp/flows"
  });
};
var getWhatsAppFlow = (options) => {
  return (options?.client ?? client).get({
    ...options,
    url: "/v1/whatsapp/flows/{flowId}"
  });
};
var updateWhatsAppFlow = (options) => {
  return (options?.client ?? client).patch({
    ...options,
    url: "/v1/whatsapp/flows/{flowId}"
  });
};
var deleteWhatsAppFlow = (options) => {
  return (options?.client ?? client).delete({
    ...options,
    url: "/v1/whatsapp/flows/{flowId}"
  });
};
var getWhatsAppFlowJson = (options) => {
  return (options?.client ?? client).get({
    ...options,
    url: "/v1/whatsapp/flows/{flowId}/json"
  });
};
var uploadWhatsAppFlowJson = (options) => {
  return (options?.client ?? client).put({
    ...options,
    url: "/v1/whatsapp/flows/{flowId}/json"
  });
};
var getWhatsAppFlowPreview = (options) => {
  return (options?.client ?? client).get({
    ...options,
    url: "/v1/whatsapp/flows/{flowId}/preview"
  });
};
var listWhatsAppFlowVersions = (options) => {
  return (options?.client ?? client).get({
    ...options,
    url: "/v1/whatsapp/flows/{flowId}/versions"
  });
};
var publishWhatsAppFlow = (options) => {
  return (options?.client ?? client).post({
    ...options,
    url: "/v1/whatsapp/flows/{flowId}/publish"
  });
};
var deprecateWhatsAppFlow = (options) => {
  return (options?.client ?? client).post({
    ...options,
    url: "/v1/whatsapp/flows/{flowId}/deprecate"
  });
};
var sendWhatsAppFlowMessage = (options) => {
  return (options?.client ?? client).post({
    ...options,
    url: "/v1/whatsapp/flows/send"
  });
};
var listWhatsAppFlowResponses = (options) => {
  return (options?.client ?? client).get({
    ...options,
    url: "/v1/whatsapp/flow-responses"
  });
};
var listContacts = (options) => {
  return (options?.client ?? client).get({
    ...options,
    url: "/v1/contacts"
  });
};
var createContact = (options) => {
  return (options?.client ?? client).post({
    ...options,
    url: "/v1/contacts"
  });
};
var getContact = (options) => {
  return (options?.client ?? client).get({
    ...options,
    url: "/v1/contacts/{contactId}"
  });
};
var updateContact = (options) => {
  return (options?.client ?? client).patch({
    ...options,
    url: "/v1/contacts/{contactId}"
  });
};
var deleteContact = (options) => {
  return (options?.client ?? client).delete({
    ...options,
    url: "/v1/contacts/{contactId}"
  });
};
var getContactChannels = (options) => {
  return (options?.client ?? client).get({
    ...options,
    url: "/v1/contacts/{contactId}/channels"
  });
};
var bulkCreateContacts = (options) => {
  return (options?.client ?? client).post({
    ...options,
    url: "/v1/contacts/bulk"
  });
};
var setContactFieldValue = (options) => {
  return (options?.client ?? client).put({
    ...options,
    url: "/v1/contacts/{contactId}/fields/{slug}"
  });
};
var clearContactFieldValue = (options) => {
  return (options?.client ?? client).delete({
    ...options,
    url: "/v1/contacts/{contactId}/fields/{slug}"
  });
};
var listCustomFields = (options) => {
  return (options?.client ?? client).get({
    ...options,
    url: "/v1/custom-fields"
  });
};
var createCustomField = (options) => {
  return (options?.client ?? client).post({
    ...options,
    url: "/v1/custom-fields"
  });
};
var updateCustomField = (options) => {
  return (options?.client ?? client).patch({
    ...options,
    url: "/v1/custom-fields/{fieldId}"
  });
};
var deleteCustomField = (options) => {
  return (options?.client ?? client).delete({
    ...options,
    url: "/v1/custom-fields/{fieldId}"
  });
};
var listBroadcasts = (options) => {
  return (options?.client ?? client).get({
    ...options,
    url: "/v1/broadcasts"
  });
};
var createBroadcast = (options) => {
  return (options?.client ?? client).post({
    ...options,
    url: "/v1/broadcasts"
  });
};
var getBroadcast = (options) => {
  return (options?.client ?? client).get({
    ...options,
    url: "/v1/broadcasts/{broadcastId}"
  });
};
var updateBroadcast = (options) => {
  return (options?.client ?? client).patch({
    ...options,
    url: "/v1/broadcasts/{broadcastId}"
  });
};
var deleteBroadcast = (options) => {
  return (options?.client ?? client).delete({
    ...options,
    url: "/v1/broadcasts/{broadcastId}"
  });
};
var sendBroadcast = (options) => {
  return (options?.client ?? client).post({
    ...options,
    url: "/v1/broadcasts/{broadcastId}/send"
  });
};
var scheduleBroadcast = (options) => {
  return (options?.client ?? client).post({
    ...options,
    url: "/v1/broadcasts/{broadcastId}/schedule"
  });
};
var cancelBroadcast = (options) => {
  return (options?.client ?? client).post({
    ...options,
    url: "/v1/broadcasts/{broadcastId}/cancel"
  });
};
var listBroadcastRecipients = (options) => {
  return (options?.client ?? client).get({
    ...options,
    url: "/v1/broadcasts/{broadcastId}/recipients"
  });
};
var addBroadcastRecipients = (options) => {
  return (options?.client ?? client).post({
    ...options,
    url: "/v1/broadcasts/{broadcastId}/recipients"
  });
};
var listWorkflows = (options) => {
  return (options?.client ?? client).get({
    ...options,
    url: "/v1/workflows"
  });
};
var createWorkflow = (options) => {
  return (options?.client ?? client).post({
    ...options,
    url: "/v1/workflows"
  });
};
var getWorkflow = (options) => {
  return (options?.client ?? client).get({
    ...options,
    url: "/v1/workflows/{workflowId}"
  });
};
var updateWorkflow = (options) => {
  return (options?.client ?? client).patch({
    ...options,
    url: "/v1/workflows/{workflowId}"
  });
};
var deleteWorkflow = (options) => {
  return (options?.client ?? client).delete({
    ...options,
    url: "/v1/workflows/{workflowId}"
  });
};
var activateWorkflow = (options) => {
  return (options?.client ?? client).post({
    ...options,
    url: "/v1/workflows/{workflowId}/activate"
  });
};
var pauseWorkflow = (options) => {
  return (options?.client ?? client).post({
    ...options,
    url: "/v1/workflows/{workflowId}/pause"
  });
};
var listWorkflowExecutions = (options) => {
  return (options?.client ?? client).get({
    ...options,
    url: "/v1/workflows/{workflowId}/executions"
  });
};
var triggerWorkflow = (options) => {
  return (options?.client ?? client).post({
    ...options,
    url: "/v1/workflows/{workflowId}/executions"
  });
};
var listWorkflowExecutionEvents = (options) => {
  return (options?.client ?? client).get({
    ...options,
    url: "/v1/workflows/{workflowId}/executions/{executionId}/events"
  });
};
var duplicateWorkflow = (options) => {
  return (options?.client ?? client).post({
    ...options,
    url: "/v1/workflows/{workflowId}/duplicate"
  });
};
var listWorkflowVersions = (options) => {
  return (options?.client ?? client).get({
    ...options,
    url: "/v1/workflows/{workflowId}/versions"
  });
};
var getWorkflowVersion = (options) => {
  return (options?.client ?? client).get({
    ...options,
    url: "/v1/workflows/{workflowId}/versions/{version}"
  });
};
var restoreWorkflowVersion = (options) => {
  return (options?.client ?? client).post({
    ...options,
    url: "/v1/workflows/{workflowId}/versions/{version}/restore"
  });
};
var listSequences = (options) => {
  return (options?.client ?? client).get({
    ...options,
    url: "/v1/sequences"
  });
};
var createSequence = (options) => {
  return (options?.client ?? client).post({
    ...options,
    url: "/v1/sequences"
  });
};
var getSequence = (options) => {
  return (options?.client ?? client).get({
    ...options,
    url: "/v1/sequences/{sequenceId}"
  });
};
var updateSequence = (options) => {
  return (options?.client ?? client).patch({
    ...options,
    url: "/v1/sequences/{sequenceId}"
  });
};
var deleteSequence = (options) => {
  return (options?.client ?? client).delete({
    ...options,
    url: "/v1/sequences/{sequenceId}"
  });
};
var activateSequence = (options) => {
  return (options?.client ?? client).post({
    ...options,
    url: "/v1/sequences/{sequenceId}/activate"
  });
};
var pauseSequence = (options) => {
  return (options?.client ?? client).post({
    ...options,
    url: "/v1/sequences/{sequenceId}/pause"
  });
};
var enrollContacts = (options) => {
  return (options?.client ?? client).post({
    ...options,
    url: "/v1/sequences/{sequenceId}/enroll"
  });
};
var unenrollContact = (options) => {
  return (options?.client ?? client).delete({
    ...options,
    url: "/v1/sequences/{sequenceId}/enroll/{contactId}"
  });
};
var listSequenceEnrollments = (options) => {
  return (options?.client ?? client).get({
    ...options,
    url: "/v1/sequences/{sequenceId}/enrollments"
  });
};
var listCommentAutomations = (options) => {
  return (options?.client ?? client).get({
    ...options,
    url: "/v1/comment-automations"
  });
};
var createCommentAutomation = (options) => {
  return (options?.client ?? client).post({
    ...options,
    url: "/v1/comment-automations"
  });
};
var getCommentAutomation = (options) => {
  return (options?.client ?? client).get({
    ...options,
    url: "/v1/comment-automations/{automationId}"
  });
};
var updateCommentAutomation = (options) => {
  return (options?.client ?? client).patch({
    ...options,
    url: "/v1/comment-automations/{automationId}"
  });
};
var deleteCommentAutomation = (options) => {
  return (options?.client ?? client).delete({
    ...options,
    url: "/v1/comment-automations/{automationId}"
  });
};
var listCommentAutomationLogs = (options) => {
  return (options?.client ?? client).get({
    ...options,
    url: "/v1/comment-automations/{automationId}/logs"
  });
};
var listAds = (options) => {
  return (options?.client ?? client).get({
    ...options,
    url: "/v1/ads"
  });
};
var listAdCampaigns = (options) => {
  return (options?.client ?? client).get({
    ...options,
    url: "/v1/ads/campaigns"
  });
};
var updateAdCampaignStatus = (options) => {
  return (options?.client ?? client).put({
    ...options,
    url: "/v1/ads/campaigns/{campaignId}/status"
  });
};
var updateAdCampaign = (options) => {
  return (options?.client ?? client).put({
    ...options,
    url: "/v1/ads/campaigns/{campaignId}"
  });
};
var deleteAdCampaign = (options) => {
  return (options?.client ?? client).delete({
    ...options,
    url: "/v1/ads/campaigns/{campaignId}"
  });
};
var bulkUpdateAdCampaignStatus = (options) => {
  return (options?.client ?? client).post({
    ...options,
    url: "/v1/ads/campaigns/bulk-status"
  });
};
var duplicateAdCampaign = (options) => {
  return (options?.client ?? client).post({
    ...options,
    url: "/v1/ads/campaigns/{campaignId}/duplicate"
  });
};
var updateAdSet = (options) => {
  return (options?.client ?? client).put({
    ...options,
    url: "/v1/ads/ad-sets/{adSetId}"
  });
};
var updateAdSetStatus = (options) => {
  return (options?.client ?? client).put({
    ...options,
    url: "/v1/ads/ad-sets/{adSetId}/status"
  });
};
var getAdTree = (options) => {
  return (options?.client ?? client).get({
    ...options,
    url: "/v1/ads/tree"
  });
};
var getAdsTimeline = (options) => {
  return (options?.client ?? client).get({
    ...options,
    url: "/v1/ads/timeline"
  });
};
var getAd = (options) => {
  return (options?.client ?? client).get({
    ...options,
    url: "/v1/ads/{adId}"
  });
};
var updateAd = (options) => {
  return (options?.client ?? client).put({
    ...options,
    url: "/v1/ads/{adId}"
  });
};
var deleteAd = (options) => {
  return (options?.client ?? client).delete({
    ...options,
    url: "/v1/ads/{adId}"
  });
};
var updateAdStatus = (options) => {
  return (options?.client ?? client).put({
    ...options,
    url: "/v1/ads/{adId}/status"
  });
};
var getCampaignAnalytics = (options) => {
  return (options?.client ?? client).get({
    ...options,
    url: "/v1/ads/campaigns/{campaignId}/analytics"
  });
};
var getAdAnalytics = (options) => {
  return (options?.client ?? client).get({
    ...options,
    url: "/v1/ads/{adId}/analytics"
  });
};
var getAdTrackingTags = (options) => {
  return (options?.client ?? client).get({
    ...options,
    url: "/v1/ads/{adId}/tracking-tags"
  });
};
var updateAdTrackingTags = (options) => {
  return (options?.client ?? client).patch({
    ...options,
    url: "/v1/ads/{adId}/tracking-tags"
  });
};
var getAdComments = (options) => {
  return (options?.client ?? client).get({
    ...options,
    url: "/v1/ads/{adId}/comments"
  });
};
var listAdsBusinessCenters = (options) => {
  return (options?.client ?? client).get({
    ...options,
    url: "/v1/ads/business-centers"
  });
};
var listAdAccounts = (options) => {
  return (options?.client ?? client).get({
    ...options,
    url: "/v1/ads/accounts"
  });
};
var updateAdAccount = (options) => {
  return (options?.client ?? client).patch({
    ...options,
    url: "/v1/ads/accounts"
  });
};
var getDsaDefaults = (options) => {
  return (options?.client ?? client).get({
    ...options,
    url: "/v1/ads/dsa-defaults"
  });
};
var getDsaRecommendations = (options) => {
  return (options?.client ?? client).get({
    ...options,
    url: "/v1/ads/dsa-recommendations"
  });
};
var boostPost = (options) => {
  return (options?.client ?? client).post({
    ...options,
    url: "/v1/ads/boost"
  });
};
var createStandaloneAd = (options) => {
  return (options?.client ?? client).post({
    ...options,
    url: "/v1/ads/create"
  });
};
var listLeads = (options) => {
  return (options?.client ?? client).get({
    ...options,
    url: "/v1/ads/leads"
  });
};
var listLeadForms = (options) => {
  return (options?.client ?? client).get({
    ...options,
    url: "/v1/ads/lead-forms"
  });
};
var createLeadForm = (options) => {
  return (options?.client ?? client).post({
    ...options,
    url: "/v1/ads/lead-forms"
  });
};
var getLeadForm = (options) => {
  return (options?.client ?? client).get({
    ...options,
    url: "/v1/ads/lead-forms/{formId}"
  });
};
var archiveLeadForm = (options) => {
  return (options?.client ?? client).delete({
    ...options,
    url: "/v1/ads/lead-forms/{formId}"
  });
};
var listFormLeads = (options) => {
  return (options?.client ?? client).get({
    ...options,
    url: "/v1/ads/lead-forms/{formId}/leads"
  });
};
var createTestLead = (options) => {
  return (options?.client ?? client).post({
    ...options,
    url: "/v1/ads/lead-forms/{formId}/test-leads"
  });
};
var searchAdInterests = (options) => {
  return (options?.client ?? client).get({
    ...options,
    url: "/v1/ads/interests"
  });
};
var searchAdTargeting = (options) => {
  return (options?.client ?? client).get({
    ...options,
    url: "/v1/ads/targeting/search"
  });
};
var estimateAdReach = (options) => {
  return (options?.client ?? client).post({
    ...options,
    url: "/v1/ads/targeting/reach-estimate"
  });
};
var getLinkedInBidPricing = (options) => {
  return (options?.client ?? client).post({
    ...options,
    url: "/v1/ads/targeting/bid-pricing"
  });
};
var getLinkedInSupplyForecast = (options) => {
  return (options?.client ?? client).post({
    ...options,
    url: "/v1/ads/targeting/supply-forecast"
  });
};
var listAdCatalogs = (options) => {
  return (options?.client ?? client).get({
    ...options,
    url: "/v1/ads/catalogs"
  });
};
var listAdCatalogProductSets = (options) => {
  return (options?.client ?? client).get({
    ...options,
    url: "/v1/ads/catalogs/{catalogId}/product-sets"
  });
};
var listAdAudiences = (options) => {
  return (options?.client ?? client).get({
    ...options,
    url: "/v1/ads/audiences"
  });
};
var createAdAudience = (options) => {
  return (options?.client ?? client).post({
    ...options,
    url: "/v1/ads/audiences"
  });
};
var getAdAudience = (options) => {
  return (options?.client ?? client).get({
    ...options,
    url: "/v1/ads/audiences/{audienceId}"
  });
};
var updateAdAudience = (options) => {
  return (options?.client ?? client).put({
    ...options,
    url: "/v1/ads/audiences/{audienceId}"
  });
};
var deleteAdAudience = (options) => {
  return (options?.client ?? client).delete({
    ...options,
    url: "/v1/ads/audiences/{audienceId}"
  });
};
var addUsersToAdAudience = (options) => {
  return (options?.client ?? client).post({
    ...options,
    url: "/v1/ads/audiences/{audienceId}/users"
  });
};
var getConversionsQuality = (options) => {
  return (options?.client ?? client).get({
    ...options,
    url: "/v1/ads/conversions/quality"
  });
};
var sendConversions = (options) => {
  return (options?.client ?? client).post({
    ...options,
    url: "/v1/ads/conversions"
  });
};
var adjustConversions = (options) => {
  return (options?.client ?? client).post({
    ...options,
    url: "/v1/ads/conversions/adjustments"
  });
};
var listConversionDestinations = (options) => {
  return (options?.client ?? client).get({
    ...options,
    url: "/v1/accounts/{accountId}/conversion-destinations"
  });
};
var createConversionDestination = (options) => {
  return (options?.client ?? client).post({
    ...options,
    url: "/v1/accounts/{accountId}/conversion-destinations"
  });
};
var getConversionDestination = (options) => {
  return (options?.client ?? client).get({
    ...options,
    url: "/v1/accounts/{accountId}/conversion-destinations/{destinationId}"
  });
};
var updateConversionDestination = (options) => {
  return (options?.client ?? client).patch({
    ...options,
    url: "/v1/accounts/{accountId}/conversion-destinations/{destinationId}"
  });
};
var deleteConversionDestination = (options) => {
  return (options?.client ?? client).delete({
    ...options,
    url: "/v1/accounts/{accountId}/conversion-destinations/{destinationId}"
  });
};
var listConversionAssociations = (options) => {
  return (options?.client ?? client).get({
    ...options,
    url: "/v1/accounts/{accountId}/conversion-destinations/{destinationId}/associations"
  });
};
var addConversionAssociations = (options) => {
  return (options?.client ?? client).post({
    ...options,
    url: "/v1/accounts/{accountId}/conversion-destinations/{destinationId}/associations"
  });
};
var removeConversionAssociations = (options) => {
  return (options?.client ?? client).delete({
    ...options,
    url: "/v1/accounts/{accountId}/conversion-destinations/{destinationId}/associations"
  });
};
var getConversionMetrics = (options) => {
  return (options?.client ?? client).get({
    ...options,
    url: "/v1/accounts/{accountId}/conversion-destinations/{destinationId}/metrics"
  });
};
var listWhatsAppConversions = (options) => {
  return (options?.client ?? client).get({
    ...options,
    url: "/v1/whatsapp/conversions"
  });
};
var sendWhatsAppConversion = (options) => {
  return (options?.client ?? client).post({
    ...options,
    url: "/v1/whatsapp/conversions"
  });
};
var createCtwaAd = (options) => {
  return (options?.client ?? client).post({
    ...options,
    url: "/v1/ads/ctwa"
  });
};
var listTrackingTags = (options) => {
  return (options?.client ?? client).get({
    ...options,
    url: "/v1/accounts/{accountId}/tracking-tags"
  });
};
var createTrackingTag = (options) => {
  return (options?.client ?? client).post({
    ...options,
    url: "/v1/accounts/{accountId}/tracking-tags"
  });
};
var getTrackingTag = (options) => {
  return (options?.client ?? client).get({
    ...options,
    url: "/v1/accounts/{accountId}/tracking-tags/{tagId}"
  });
};
var updateTrackingTag = (options) => {
  return (options?.client ?? client).patch({
    ...options,
    url: "/v1/accounts/{accountId}/tracking-tags/{tagId}"
  });
};
var listTrackingTagSharedAccounts = (options) => {
  return (options?.client ?? client).get({
    ...options,
    url: "/v1/accounts/{accountId}/tracking-tags/{tagId}/shared-accounts"
  });
};
var addTrackingTagSharedAccount = (options) => {
  return (options?.client ?? client).post({
    ...options,
    url: "/v1/accounts/{accountId}/tracking-tags/{tagId}/shared-accounts"
  });
};
var removeTrackingTagSharedAccount = (options) => {
  return (options?.client ?? client).delete({
    ...options,
    url: "/v1/accounts/{accountId}/tracking-tags/{tagId}/shared-accounts"
  });
};
var getTrackingTagStats = (options) => {
  return (options?.client ?? client).get({
    ...options,
    url: "/v1/accounts/{accountId}/tracking-tags/{tagId}/stats"
  });
};

// src/errors.ts
var ZernioApiError = class _ZernioApiError extends Error {
  constructor(message, statusCode, code, details) {
    super(message);
    this.name = "ZernioApiError";
    this.statusCode = statusCode;
    this.code = code;
    this.details = details;
    if (Error.captureStackTrace) {
      Error.captureStackTrace(this, _ZernioApiError);
    }
  }
  /**
   * Check if this is a rate limit error
   */
  isRateLimited() {
    return this.statusCode === 429;
  }
  /**
   * Check if this is an authentication error
   */
  isAuthError() {
    return this.statusCode === 401;
  }
  /**
   * Check if this is a permission/access error
   */
  isForbidden() {
    return this.statusCode === 403;
  }
  /**
   * Check if this is a not found error
   */
  isNotFound() {
    return this.statusCode === 404;
  }
  /**
   * Check if this is a validation error
   */
  isValidationError() {
    return this.statusCode === 400;
  }
  /**
   * Check if this is a payment required error
   */
  isPaymentRequired() {
    return this.statusCode === 402;
  }
};
var LateApiError = ZernioApiError;
var RateLimitError = class extends ZernioApiError {
  constructor(message, limit, remaining, resetAt) {
    super(message, 429, "rate_limit_exceeded");
    this.name = "RateLimitError";
    this.limit = limit;
    this.remaining = remaining;
    this.resetAt = resetAt;
  }
  /**
   * Get seconds until rate limit resets
   */
  getSecondsUntilReset() {
    if (!this.resetAt) return void 0;
    return Math.max(0, Math.ceil((this.resetAt.getTime() - Date.now()) / 1e3));
  }
};
var ValidationError = class extends ZernioApiError {
  constructor(message, fields) {
    super(message, 400, "validation_error", { fields });
    this.name = "ValidationError";
    this.fields = fields;
  }
};
function parseApiError(response, body) {
  const message = body?.error || body?.message || response.statusText || "Unknown error";
  const code = body?.code;
  const details = body?.details;
  if (response.status === 429) {
    const limit = response.headers.get("X-RateLimit-Limit");
    const remaining = response.headers.get("X-RateLimit-Remaining");
    const reset = response.headers.get("X-RateLimit-Reset");
    return new RateLimitError(
      message,
      limit ? parseInt(limit, 10) : void 0,
      remaining ? parseInt(remaining, 10) : void 0,
      reset ? new Date(parseInt(reset, 10) * 1e3) : void 0
    );
  }
  if (response.status === 400 && details?.fields) {
    return new ValidationError(message, details.fields);
  }
  return new ZernioApiError(message, response.status, code, details);
}

// src/client.ts
var Zernio = class {
  /**
   * Create a new Zernio API client.
   *
   * @param options - Configuration options for the client
   */
  constructor(options = {}) {
    /**
     * validate API
     */
    this.validate = {
      validatePostLength,
      validatePost,
      validateMedia,
      validateSubreddit
    };
    /**
     * Analytics API - Get performance metrics
     */
    this.analytics = {
      getAnalytics,
      getYouTubeChannelInsights,
      getLinkedInOrgAggregateAnalytics,
      getTikTokAccountInsights,
      getYouTubeDailyViews,
      getYouTubeVideoRetention,
      getFacebookPageInsights,
      getInstagramAccountInsights,
      getInstagramFollowerHistory,
      getInstagramDemographics,
      getYouTubeDemographics,
      getDailyMetrics,
      getBestTimeToPost,
      getContentDecay,
      getPostingFrequency,
      getPostTimeline,
      getGoogleBusinessPerformance,
      getGoogleBusinessSearchKeywords,
      syncExternalPosts,
      getLinkedInAggregateAnalytics,
      getLinkedInPostAnalytics,
      getLinkedInPostReactions,
      getFacebookPostReactions
    };
    /**
     * inboxanalytics API
     */
    this.inboxanalytics = {
      getInboxVolume,
      getInboxHeatmap,
      getInboxSourceBreakdown,
      getInboxResponseTime,
      getInboxTopAccounts,
      listInboxConversationAnalytics,
      getInboxConversationAnalytics
    };
    /**
     * Account Groups API - Organize accounts into groups
     */
    this.accountGroups = {
      listAccountGroups,
      createAccountGroup,
      updateAccountGroup,
      deleteAccountGroup
    };
    /**
     * Media API - Upload and manage media files
     */
    this.media = {
      getMediaPresignedUrl
    };
    /**
     * Reddit API - Search and feed
     */
    this.reddit = {
      searchReddit,
      getRedditFeed
    };
    /**
     * Usage API - Get usage statistics
     */
    this.usage = {
      getBilling,
      getXApiPricing,
      getUsage,
      getUsageStats,
      getCallsUsage,
      getSmsUsage
    };
    /**
     * Posts API - Create, schedule, and manage social media posts
     */
    this.posts = {
      listPosts,
      createPost,
      getPost,
      updatePost,
      deletePost,
      bulkUploadPosts,
      retryPost,
      unpublishPost,
      editPost,
      updatePostMetadata
    };
    /**
     * Users API - User management
     */
    this.users = {
      listUsers,
      getUser
    };
    /**
     * Profiles API - Manage workspace profiles
     */
    this.profiles = {
      listProfiles,
      createProfile,
      getProfile,
      updateProfile,
      deleteProfile
    };
    /**
     * Accounts API - Manage connected social media accounts
     */
    this.accounts = {
      listAccounts,
      getFollowerStats,
      updateAccount,
      moveAccountToProfile,
      deleteAccount,
      getAllAccountsHealth,
      getAccountHealth,
      getTikTokCreatorInfo,
      getGoogleBusinessReviews,
      batchGetGoogleBusinessReviews,
      replyToGoogleBusinessReview,
      deleteGoogleBusinessReviewReply,
      getLinkedInMentions
    };
    /**
     * API Keys API - Manage API keys
     */
    this.apiKeys = {
      listApiKeys,
      createApiKey,
      deleteApiKey
    };
    /**
     * Invites API - Team invitations
     */
    this.invites = {
      createInviteToken
    };
    /**
     * Connect API - OAuth connection flows
     */
    this.connect = {
      getConnectUrl,
      handleOAuthCallback,
      connectAds,
      configureTikTokAdsBrandIdentity,
      getPendingOAuthData,
      connectWhatsAppCredentials,
      listWhatsAppPhoneNumbers,
      completeWhatsAppPhoneSelection,
      getFacebookPages,
      updateFacebookPage,
      getLinkedInOrganizations,
      updateLinkedInOrganization,
      getPinterestBoards,
      updatePinterestBoards,
      createPinterestBoard,
      getYoutubePlaylists,
      updateYoutubeDefaultPlaylist,
      getGmbLocations,
      updateGmbLocation,
      getRedditSubreddits,
      updateRedditSubreddits,
      getSubredditRules,
      voteRedditThing,
      getRedditFlairs,
      setRedditPostFlair,
      facebook: {
        listFacebookPages,
        selectFacebookPage
      },
      googleBusiness: {
        listGoogleBusinessLocations,
        selectGoogleBusinessLocation
      },
      linkedin: {
        listLinkedInOrganizations,
        selectLinkedInOrganization
      },
      pinterest: {
        listPinterestBoardsForSelection,
        selectPinterestBoard
      },
      snapchat: {
        listSnapchatProfiles,
        selectSnapchatProfile
      },
      bluesky: {
        connectBlueskyCredentials
      },
      telegram: {
        getTelegramConnectStatus,
        initiateTelegramConnect,
        completeTelegramConnect
      }
    };
    /**
     * gmbverifications API
     */
    this.gmbverifications = {
      getGoogleBusinessVerifications,
      startGoogleBusinessVerification,
      fetchGoogleBusinessVerificationOptions,
      completeGoogleBusinessVerification
    };
    /**
     * gmbfoodmenus API
     */
    this.gmbfoodmenus = {
      getGoogleBusinessFoodMenus,
      updateGoogleBusinessFoodMenus
    };
    /**
     * gmblocationdetails API
     */
    this.gmblocationdetails = {
      getGoogleBusinessLocationDetails,
      updateGoogleBusinessLocationDetails
    };
    /**
     * gmbmedia API
     */
    this.gmbmedia = {
      listGoogleBusinessMedia,
      createGoogleBusinessMedia,
      deleteGoogleBusinessMedia
    };
    /**
     * gmbattributes API
     */
    this.gmbattributes = {
      getGmbAttributeMetadata,
      getGoogleBusinessAttributes,
      updateGoogleBusinessAttributes
    };
    /**
     * gmbplaceactions API
     */
    this.gmbplaceactions = {
      listGoogleBusinessPlaceActions,
      createGoogleBusinessPlaceAction,
      deleteGoogleBusinessPlaceAction,
      updateGoogleBusinessPlaceAction
    };
    /**
     * gmbservices API
     */
    this.gmbservices = {
      getGoogleBusinessServices,
      updateGoogleBusinessServices
    };
    /**
     * instagram API
     */
    this.instagram = {
      listInstagramStories,
      getInstagramPublishingLimit,
      getInstagramStoryInsights
    };
    /**
     * discord API
     */
    this.discord = {
      getDiscordSettings,
      updateDiscordSettings,
      getDiscordChannels,
      sendDiscordDirectMessage,
      listDiscordGuildRoles,
      createDiscordGuildRole,
      editDiscordGuildRole,
      deleteDiscordGuildRole,
      listDiscordGuildMembers,
      addDiscordMemberRole,
      removeDiscordMemberRole,
      deleteDiscordMessage,
      crosspostDiscordMessage,
      createDiscordThread,
      listDiscordPinnedMessages,
      pinDiscordMessage,
      unpinDiscordMessage,
      listDiscordScheduledEvents,
      createDiscordScheduledEvent,
      getDiscordScheduledEvent,
      updateDiscordScheduledEvent,
      deleteDiscordScheduledEvent
    };
    /**
     * Queue API - Manage posting queue
     */
    this.queue = {
      listQueueSlots,
      createQueueSlot,
      updateQueueSlot,
      deleteQueueSlot,
      previewQueue,
      getNextQueueSlot
    };
    /**
     * Webhooks API - Configure event webhooks
     */
    this.webhooks = {
      getWebhookSettings,
      createWebhookSettings,
      updateWebhookSettings,
      deleteWebhookSettings,
      getWebhookLogs,
      testWebhook
    };
    /**
     * Logs API - Publishing logs
     */
    this.logs = {
      listLogs
    };
    /**
     * messages API
     */
    this.messages = {
      listInboxConversations,
      createInboxConversation,
      searchInboxConversations,
      getInboxConversation,
      updateInboxConversation,
      getInboxConversationMessages,
      sendInboxMessage,
      editInboxMessage,
      deleteInboxMessage,
      sendTypingIndicator,
      markConversationRead,
      addMessageReaction,
      removeMessageReaction,
      uploadMediaDirect
    };
    /**
     * accountsettings API
     */
    this.accountsettings = {
      getMessengerMenu,
      setMessengerMenu,
      deleteMessengerMenu,
      getInstagramIceBreakers,
      setInstagramIceBreakers,
      deleteInstagramIceBreakers,
      getTelegramCommands,
      setTelegramCommands,
      deleteTelegramCommands
    };
    /**
     * comments API
     */
    this.comments = {
      listInboxComments,
      getInboxPostComments,
      replyToInboxPost,
      deleteInboxComment,
      editInboxComment,
      setCommentModeration,
      hideInboxComment,
      unhideInboxComment,
      likeInboxComment,
      unlikeInboxComment,
      sendPrivateReplyToComment
    };
    /**
     * twitterengagement API
     */
    this.twitterengagement = {
      retweetPost,
      undoRetweet,
      bookmarkPost,
      removeBookmark,
      followUser,
      unfollowUser
    };
    /**
     * mentions API
     */
    this.mentions = {
      listInboxMentions,
      replyToMention
    };
    /**
     * reviews API
     */
    this.reviews = {
      listInboxReviews,
      replyToInboxReview,
      deleteInboxReviewReply
    };
    /**
     * whatsapp API
     */
    this.whatsapp = {
      getWhatsAppTemplates,
      createWhatsAppTemplate,
      getWhatsAppTemplate,
      updateWhatsAppTemplate,
      deleteWhatsAppTemplate,
      getWhatsAppBusinessProfile,
      updateWhatsAppBusinessProfile,
      uploadWhatsAppProfilePhoto,
      getWhatsAppDisplayName,
      updateWhatsAppDisplayName,
      getWhatsappBusinessUsername,
      setWhatsappBusinessUsername,
      deleteWhatsappBusinessUsername,
      getWhatsappBusinessUsernameSuggestions,
      getWhatsAppBlockStatus,
      getWhatsAppBlockedUsers,
      blockWhatsAppUsers,
      unblockWhatsAppUsers,
      getWhatsAppDataset,
      createWhatsAppDataset,
      listWhatsAppGroupChats,
      createWhatsAppGroupChat,
      getWhatsAppGroupChat,
      updateWhatsAppGroupChat,
      deleteWhatsAppGroupChat,
      addWhatsAppGroupParticipants,
      removeWhatsAppGroupParticipants,
      createWhatsAppGroupInviteLink,
      listWhatsAppGroupJoinRequests,
      approveWhatsAppGroupJoinRequests,
      rejectWhatsAppGroupJoinRequests,
      listWhatsAppConversions,
      sendWhatsAppConversion
    };
    /**
     * whatsappcalling API
     */
    this.whatsappcalling = {
      getWhatsAppCallingConfig,
      enableWhatsAppCallingLegacy,
      updateWhatsAppCallingLegacy,
      disableWhatsAppCallingLegacy,
      getWhatsAppCallPermissions,
      initiateWhatsAppCall,
      listWhatsAppCalls,
      getWhatsAppCall,
      getWhatsAppCallRecording,
      getWhatsAppCallEstimate,
      getWhatsAppCalling,
      enableWhatsAppCalling,
      updateWhatsAppCalling,
      disableWhatsAppCalling
    };
    /**
     * calls API
     */
    this.calls = {
      listCalls,
      getCall,
      getCallRecording
    };
    /**
     * voice API
     */
    this.voice = {
      createVoiceCall,
      listVoiceCalls,
      getVoiceCall,
      endVoiceCall,
      getVoiceCallRecording,
      transferVoiceCall,
      getVoiceCallEstimate,
      createVoiceWebSession,
      dialVoiceWebCall,
      enableVoiceOnNumber,
      disableVoiceOnNumber
    };
    /**
     * sms API
     */
    this.sms = {
      sendSms,
      lookupSmsNumber,
      listSmsOptOuts,
      startSmsRegistration,
      listSmsRegistrations,
      deactivateSmsRegistration,
      getSmsRegistration,
      verifySmsRegistrationOtp,
      resendSmsRegistrationOtp,
      appealSmsRegistration,
      uploadSmsOptInProofFile,
      uploadSmsOptInProof,
      shareSmsRegistration,
      enableSmsOnNumber,
      disableSmsOnNumber,
      reuseSmsRegistrationForNumber
    };
    /**
     * whatsapptemplates API
     */
    this.whatsapptemplates = {
      getWhatsAppLibraryTemplate
    };
    /**
     * whatsappphonenumbers API
     */
    this.whatsappphonenumbers = {
      getWhatsAppNumberInfo,
      getWhatsAppPhoneNumbers,
      purchaseWhatsAppPhoneNumber,
      listWhatsAppNumberCountries,
      searchAvailableWhatsAppNumbers,
      checkWhatsAppNumberAvailability,
      getWhatsAppNumberKycForm,
      submitWhatsAppNumberKyc,
      uploadWhatsAppNumberKycDocument,
      validateWhatsAppNumberKycAddress,
      createWhatsAppNumberKycLink,
      getWhatsAppNumberRemediation,
      remediateWhatsAppNumber,
      getWhatsAppPhoneNumber,
      releaseWhatsAppPhoneNumber
    };
    /**
     * phonenumbers API
     */
    this.phonenumbers = {
      listPhoneNumbers,
      getPhoneNumber,
      releasePhoneNumber,
      purchasePhoneNumber,
      listPhoneNumberCountries,
      searchAvailablePhoneNumbers,
      checkPhoneNumberAvailability,
      getPhoneNumberKycForm,
      submitPhoneNumberKyc,
      uploadPhoneNumberKycDocument,
      validatePhoneNumberKycAddress,
      createPhoneNumberKycLink,
      createPhoneNumberPortIn,
      listPhoneNumberPortIns,
      checkPhoneNumberPortability,
      uploadPhoneNumberPortInDocument,
      cancelPhoneNumberPortIn,
      reviewPhoneNumberKycPacket,
      getPhoneNumberRemediation,
      remediatePhoneNumber
    };
    /**
     * whatsappsandbox API
     */
    this.whatsappsandbox = {
      listWhatsAppSandboxSessions,
      createWhatsAppSandboxSession,
      deleteWhatsAppSandboxSession
    };
    /**
     * whatsappflows API
     */
    this.whatsappflows = {
      listWhatsAppFlows,
      createWhatsAppFlow,
      getWhatsAppFlow,
      updateWhatsAppFlow,
      deleteWhatsAppFlow,
      getWhatsAppFlowJson,
      uploadWhatsAppFlowJson,
      getWhatsAppFlowPreview,
      listWhatsAppFlowVersions,
      publishWhatsAppFlow,
      deprecateWhatsAppFlow,
      sendWhatsAppFlowMessage,
      listWhatsAppFlowResponses
    };
    /**
     * contacts API
     */
    this.contacts = {
      listContacts,
      createContact,
      getContact,
      updateContact,
      deleteContact,
      getContactChannels,
      bulkCreateContacts
    };
    /**
     * customfields API
     */
    this.customfields = {
      setContactFieldValue,
      clearContactFieldValue,
      listCustomFields,
      createCustomField,
      updateCustomField,
      deleteCustomField
    };
    /**
     * broadcasts API
     */
    this.broadcasts = {
      listBroadcasts,
      createBroadcast,
      getBroadcast,
      updateBroadcast,
      deleteBroadcast,
      sendBroadcast,
      scheduleBroadcast,
      cancelBroadcast,
      listBroadcastRecipients,
      addBroadcastRecipients
    };
    /**
     * workflows API
     */
    this.workflows = {
      listWorkflows,
      createWorkflow,
      getWorkflow,
      updateWorkflow,
      deleteWorkflow,
      activateWorkflow,
      pauseWorkflow,
      listWorkflowExecutions,
      triggerWorkflow,
      listWorkflowExecutionEvents,
      duplicateWorkflow,
      listWorkflowVersions,
      getWorkflowVersion,
      restoreWorkflowVersion
    };
    /**
     * sequences API
     */
    this.sequences = {
      listSequences,
      createSequence,
      getSequence,
      updateSequence,
      deleteSequence,
      activateSequence,
      pauseSequence,
      enrollContacts,
      unenrollContact,
      listSequenceEnrollments
    };
    /**
     * commentautomations API
     */
    this.commentautomations = {
      listCommentAutomations,
      createCommentAutomation,
      getCommentAutomation,
      updateCommentAutomation,
      deleteCommentAutomation,
      listCommentAutomationLogs
    };
    /**
     * ads API
     */
    this.ads = {
      listAds,
      getAd,
      updateAd,
      deleteAd,
      updateAdStatus,
      getCampaignAnalytics,
      getAdAnalytics,
      getAdTrackingTags,
      updateAdTrackingTags,
      getAdComments,
      listAdsBusinessCenters,
      listAdAccounts,
      updateAdAccount,
      getDsaDefaults,
      getDsaRecommendations,
      boostPost,
      createStandaloneAd,
      listLeads,
      listLeadForms,
      createLeadForm,
      getLeadForm,
      archiveLeadForm,
      listFormLeads,
      createTestLead,
      searchAdInterests,
      searchAdTargeting,
      estimateAdReach,
      getLinkedInBidPricing,
      getLinkedInSupplyForecast,
      listAdCatalogs,
      listAdCatalogProductSets,
      getConversionsQuality,
      sendConversions,
      adjustConversions,
      listConversionDestinations,
      createConversionDestination,
      getConversionDestination,
      updateConversionDestination,
      deleteConversionDestination,
      listConversionAssociations,
      addConversionAssociations,
      removeConversionAssociations,
      getConversionMetrics,
      createCtwaAd
    };
    /**
     * adcampaigns API
     */
    this.adcampaigns = {
      listAdCampaigns,
      updateAdCampaignStatus,
      updateAdCampaign,
      deleteAdCampaign,
      bulkUpdateAdCampaignStatus,
      duplicateAdCampaign,
      updateAdSet,
      updateAdSetStatus,
      getAdTree,
      getAdsTimeline
    };
    /**
     * adaudiences API
     */
    this.adaudiences = {
      listAdAudiences,
      createAdAudience,
      getAdAudience,
      updateAdAudience,
      deleteAdAudience,
      addUsersToAdAudience
    };
    /**
     * trackingtags API
     */
    this.trackingtags = {
      listTrackingTags,
      createTrackingTag,
      getTrackingTag,
      updateTrackingTag,
      listTrackingTagSharedAccounts,
      addTrackingTagSharedAccount,
      removeTrackingTagSharedAccount,
      getTrackingTagStats
    };
    const apiKey = options.apiKey ?? process.env["ZERNIO_API_KEY"] ?? process.env["LATE_API_KEY"];
    if (!apiKey) {
      throw new ZernioApiError(
        "The ZERNIO_API_KEY environment variable is missing or empty; either provide it, or instantiate the Zernio client with an apiKey option, like new Zernio({ apiKey: 'sk_...' }). LATE_API_KEY is also supported for backwards compatibility.",
        401,
        "missing_api_key"
      );
    }
    this.apiKey = apiKey;
    this.baseURL = options.baseURL ?? "https://zernio.com/api";
    this._options = options;
    client.setConfig({
      baseUrl: this.baseURL,
      headers: {
        "User-Agent": `zernio-node/${package_default.version}`,
        ...options.defaultHeaders ?? {}
      }
    });
    client.interceptors.request.use((request) => {
      request.headers.set("Authorization", `Bearer ${this.apiKey}`);
      return request;
    });
    client.interceptors.response.use(async (response) => {
      if (!response.ok) {
        let body;
        try {
          body = await response.clone().json();
        } catch {
        }
        throw parseApiError(response, body);
      }
      return response;
    });
  }
};
var Late = Zernio;
export {
  Late,
  LateApiError,
  RateLimitError,
  ValidationError,
  Zernio,
  ZernioApiError,
  Zernio as default,
  parseApiError
};
